diff -upr a/lib/rdk/getDeviceId.sh b/lib/rdk/getDeviceId.sh
--- a/lib/rdk/getDeviceId.sh	2025-07-01 01:25:23.865443317 +0300
+++ b/lib/rdk/getDeviceId.sh	2025-07-01 01:29:32.397412506 +0300
@@ -74,4 +74,5 @@ else
 fi
 
 echo "{ \"deviceId\" : \"${deviceId}\", \"partnerId\" : \"${partnerId}\" }"
-t2ValNotify "SYST_INFO_PartnerId" "$partnerId"
+date > /dev/stderr
+time t2ValNotify "SYST_INFO_PartnerId" "$partnerId"
