Date: Jan 06, 2023 1:30 PM
Subject: Fix undeclared variable/function errors in pxcore-libnode when compiled with gcc > 11.0
Source: COMCAST
Upstream-Status: Pending
Signed-off-by: Aravindan NarasimhapuramChakravarthy <aravindan_narasimhapuramchakravarthy@comcast.com>

Index: git/src/pxColorNames.cpp
===================================================================
--- git.orig/src/pxColorNames.cpp
+++ git/src/pxColorNames.cpp
@@ -21,7 +21,7 @@
 
 #include <map>
 #include <string.h> // for strlen()
-
+#include <stdio.h> //for sscanf()
 #include "pxColorNames.h"
 
 
Index: git/src/pxUtil.cpp
===================================================================
--- git.orig/src/pxUtil.cpp
+++ git/src/pxUtil.cpp
@@ -62,6 +62,25 @@
 
 pxImageType getImageType( const uint8_t* data, size_t len ); //fwd
 
+// Declarations for gcc-11 compatibility
+//png_get_first_frame_is_hidden()
+png_byte PNGAPI png_get_first_frame_is_hidden(png_structp png_ptr, png_infop info_ptr);
+//png_get_acTL()
+png_uint_32 PNGAPI png_get_acTL(png_structp png_ptr, png_infop info_ptr, png_uint_32 *num_frames, png_uint_32 *num_plays);
+//png_read_frame_head()
+void PNGAPI png_read_frame_head(png_structp png_ptr, png_infop info_ptr);
+//png_get_next_frame_fcTL()
+png_uint_32 PNGAPI png_get_next_frame_fcTL(png_structp png_ptr, png_infop info_ptr,
+		                           png_uint_32 *width, png_uint_32 *height,
+			                   png_uint_32 *x_offset, png_uint_32 *y_offset,
+					   png_uint_16 *delay_num, png_uint_16 *delay_den,
+					   png_byte *dispose_op, png_byte *blend_op);
+#define PNG_INFO_acTL              0x10000U
+#define PNG_BLEND_OP_SOURCE        0x00U
+#define PNG_DISPOSE_OP_PREVIOUS    0x02U
+#define PNG_DISPOSE_OP_BACKGROUND  0x01U
+#define PNG_BLEND_OP_OVER          0x01U
+
 class NSVGrasterizerEx
 {
   public:
