diff --git a/media/server/gstplayer/source/tasks/generic/Stop.cpp b/media/server/gstplayer/source/tasks/generic/Stop.cpp
index 3f82478b5..d9b379e72 100644
--- a/media/server/gstplayer/source/tasks/generic/Stop.cpp
+++ b/media/server/gstplayer/source/tasks/generic/Stop.cpp
@@ -21,6 +21,7 @@
 #include "GenericPlayerContext.h"
 #include "IGstGenericPlayerPrivate.h"
 #include "RialtoServerLogging.h"
+#include <malloc.h>
 
 namespace firebolt::rialto::server::tasks::generic
 {
@@ -44,5 +45,6 @@ void Stop::execute() const
         streamInfo.second.isDataNeeded = false;
     }
     RIALTO_SERVER_LOG_MIL("State change to NULL requested");
+    malloc_trim(0);
 }
 } // namespace firebolt::rialto::server::tasks::generic
diff --git a/media/server/gstplayer/source/tasks/webAudio/Stop.cpp b/media/server/gstplayer/source/tasks/webAudio/Stop.cpp
index 56d01b7ae..4258660a3 100644
--- a/media/server/gstplayer/source/tasks/webAudio/Stop.cpp
+++ b/media/server/gstplayer/source/tasks/webAudio/Stop.cpp
@@ -21,6 +21,7 @@
 #include "IGstWebAudioPlayerPrivate.h"
 #include "RialtoServerLogging.h"
 #include "WebAudioPlayerContext.h"
+#include <malloc.h>
 
 namespace firebolt::rialto::server::tasks::webaudio
 {
@@ -39,5 +40,6 @@ void Stop::execute() const
     RIALTO_SERVER_LOG_DEBUG("Executing Stop");
     m_player.changePipelineState(GST_STATE_NULL);
     RIALTO_SERVER_LOG_MIL("State change to NULL requested for webaudio pipeline");
+    malloc_trim(0);
 }
 } // namespace firebolt::rialto::server::tasks::webaudio
