diff --git a/Source/plugins/CMakeLists.txt b/Source/plugins/CMakeLists.txt
index e0ca1fdd9..0b59a82a5 100644
--- a/Source/plugins/CMakeLists.txt
+++ b/Source/plugins/CMakeLists.txt
@@ -187,7 +187,7 @@ InstallPackageConfig(
         DESCRIPTION "Basic library to realize the glue between a proprietary plugin and the framework host.")

 InstallCMakeConfig(
-        EXTRA_DEPENDENCIES "ConfigGenerator"
+        EXTRA_DEPENDENCIES "Thunder;ConfigGenerator"
         TARGETS ${TARGET})

 InstallCMakeConfig(TARGETS ${TARGET_PROXYSTUBS})
diff --git a/cmake/project.cmake.in b/cmake/project.cmake.in
index b50c42dc9..5f25e59f7 100644
--- a/cmake/project.cmake.in
+++ b/cmake/project.cmake.in
@@ -161,7 +161,7 @@ endif()

 if(TOOLS_SYSROOT)
     list(APPEND CMAKE_MODULE_PATH
-        "${TOOLS_SYSROOT}/@CMAKE_INSTALL_PREFIX@/include/${NAMESPACE}/Modules")
+        "${TOOLS_SYSROOT}/@CMAKE_INSTALL_PREFIX@/share/cmake")
 endif()

 list(REMOVE_DUPLICATES CMAKE_MODULE_PATH)
