diff --git a/Source/core/Trace.h b/Source/core/Trace.h
index 71870ca36..be97e20d0 100644
--- a/Source/core/Trace.h
+++ b/Source/core/Trace.h
@@ -98,7 +98,6 @@ namespace Thunder {

 #if defined(CORE_TRACE_NOT_ALLOWED) && !defined(__WINDOWS__)
 #define TRACE_FORMATTING(fmt, ...)                                                                            \
-    _Pragma ("GCC warning \"Using 'TRACE_Lx' outside of Thunder Core is deprecated\"")                        \
     TRACE_FORMATTING_IMPL(fmt, ##__VA_ARGS__)
 #else
 #define TRACE_FORMATTING(fmt, ...)
