Date:  Apr 30 14:49:57 2024
From: Thamim Razith
Subject: R4.4 change module path
Source: COMCAST
Upstream-Status: Pending
Signed-off-by: Thamim Razith <tabbas651@cable.comcast.com>

Index: git/CMakeLists.txt
===================================================================
--- git.orig/CMakeLists.txt
+++ git/CMakeLists.txt
@@ -32,11 +32,13 @@ if(NOT DEFINED GENERIC_CMAKE_MODULE_PATH
     else()
         set(NAMESPACE WPEFramework)
     endif()
-    if(DEFINED ENV{STAGING_DIR})
-        set(GENERIC_CMAKE_MODULE_PATH "$ENV{STAGING_DIR}/usr/include/${NAMESPACE}/Modules")
-    else()
-        set(GENERIC_CMAKE_MODULE_PATH "${CMAKE_SYSROOT}${CMAKE_INSTALL_PREFIX}/include/${NAMESPACE}/Modules")
-    endif()
+    find_path(GENERIC_CMAKE_MODULE_PATH CMake.cmake PATHS
+                "/usr/local/share/cmake/Modules/"
+                "/usr/share/cmake/Modules/"
+                "/usr/local/share/cmake-${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}/Modules/"
+                "/usr/share/cmake-${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}/Modules/"
+             )
+    message(STATUS "GENERIC_CMAKE_MODULE_PATH : ${GENERIC_CMAKE_MODULE_PATH}")
 endif()
 
 configure_file( "${CMAKE_CURRENT_SOURCE_DIR}/cmake/FindProxyStubGenerator.cmake.in"
