diff --git a/recipes-thirdparty/tr69hostif/tr69hostif_git.bb b/recipes-thirdparty/tr69hostif/tr69hostif_git.bb
index 0d8efa995..0b5bde65d 100644
--- a/recipes-thirdparty/tr69hostif/tr69hostif_git.bb
+++ b/recipes-thirdparty/tr69hostif/tr69hostif_git.bb
@@ -4,9 +4,9 @@ SECTION = "console/utils"
 LICENSE = "Apache-2.0"
 LIC_FILES_CHKSUM = "file://LICENSE;md5=76ae13a6bce633447ea2284294f073c2"

-SRCREV = "fd993ccb734d0ff26a0a890e1bd46c3c6b39cb23"
+SRCREV = "0a1feed39f98cc216f08f8d7c068668480cd904b"

-SRC_URI = "${CMF_GITHUB_ROOT}/tr69hostif;${CMF_GITHUB_SRC_URI_SUFFIX};name=tr69hostif"
+SRC_URI = "${CMF_GITHUB_ROOT}/tr69hostif;${CMF_GITHUB_SRC_URI_SUFFIX};branch=develop;name=tr69hostif"
 PV = "1.5.3"
 PR = "r0"
 PACKAGE_ARCH = "${MIDDLEWARE_ARCH}"
@@ -85,6 +85,7 @@ CXXFLAGS += " -DFETCH_PRODUCTCLASS_FROM_MFRLIB "
 CXXFLAGS += " -Wall -Werror -Wno-cpp"

 DEPENDS += " rbus "
+DEPENDS += " opentelemetry-cpp "

 LDFLAGS:append = " -lrbus "
 CXXFLAGS:append = " -I${includedir}/rbus "
@@ -92,6 +93,8 @@ CXXFLAGS:append = " -I${includedir}/rbus "
 RDEPENDS:${PN} += "bash libsoup"
 RDEPENDS:${PN} += "${PN}-conf"

+RDEPENDS:${PN}:append = " rdk-otel-collector "
+
 RDEPENDS:${PN}:append = " ${@bb.utils.contains('DISTRO_FEATURES', 'ENABLE_NETWORKMANAGER', '', 'netsrvmgr', d)}"
 EXTRA_OECONF += "--disable-silent-rules --enable-InterfaceStack --enable-IPv6 --enable-notification --enable-yocto --enable-SpeedTest"
 EXTRA_OECONF += " --enable-parodus"
@@ -102,7 +105,7 @@ EXTRA_OECONF:append = " ${@bb.utils.contains('DISTRO_FEATURES', 'systemd', '--en
 EXTRA_OECONF:append = " ${@bb.utils.contains('DISTRO_FEATURES', 'NEW_HTTP_SERVER_DISABLE', '--disable-new-http-server', '', d)}"

 PACKAGECONFIG ??= ""
-PACKAGECONFIG[xre] = "--enable-xre,,"
+#PACKAGECONFIG[xre] = "--enable-xre,,"
 PACKAGECONFIG[moca] = "--enable-moca,,virtual/mocadriver"
 PACKAGECONFIG[moca2] = "--enable-moca2,,virtual/mocadriver"
 PACKAGECONFIG:append = " ${@bb.utils.contains('DISTRO_FEATURES', 'bluetooth','bluetooth', '',d)}"

