Date: Fri Mar 7 10:34:00 2025
From: Punam Kumbhar
Subject: Thunder_service_dependency
Source: COMCAST
Upstream-Status: Pending
Signed-off-by: Punam Kumbhar <punam_kumbhar@cable.comcast.com>

diff --git a/systemd/ripple.service b/systemd/ripple.service
index 63801c99..9f94332a 100644
--- a/systemd/ripple.service
+++ b/systemd/ripple.service
@@ -1,6 +1,7 @@
 [Unit]
 Description=Firebolt reference implementation
-After=wpeframework.service
+Requires= wpeframework.service wpeframework-system.service wpeframework-powermanager.service wpeframework-persistentstore.service wpeframework-network.service wpeframework-displaysettings.service
+After=wpeframework.service wpeframework-system.service wpeframework-powermanager.service wpeframework-displaysettings.service wpeframework-persistentstore.service wpeframework-network.service wpeframework-displaysettings.service
 
 [Service]
 Type=notify
