From c439777346789e815a86d663c9b2ffae758769cc Mon Sep 17 00:00:00 2001
From: Krishna Priya Kanagaraj <krishnapriya_kanagaraj@comcast.com>
Date: Wed, 13 May 2026 08:13:42 +0000
Subject: [PATCH] implementing OCDM Interface for robustnesslevel

---
 .../graphics/gstreamer/eme/CDMThunder.cpp     | 27 +++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/Source/WebCore/platform/graphics/gstreamer/eme/CDMThunder.cpp b/Source/WebCore/platform/graphics/gstreamer/eme/CDMThunder.cpp
index 139a26be65c7..8a226b065ed2 100644
--- a/Source/WebCore/platform/graphics/gstreamer/eme/CDMThunder.cpp
+++ b/Source/WebCore/platform/graphics/gstreamer/eme/CDMThunder.cpp
@@ -171,6 +171,33 @@ bool CDMPrivateThunder::supportsConfiguration(const CDMKeySystemConfiguration& c
 
 Vector<AtomString> CDMPrivateThunder::supportedRobustnesses() const
 {
+#if THUNDER_HAS_OCDM_SUPPORTED_ROBUSTNESS
+    Vector<AtomString> robustnesses = { emptyAtom() };
+    
+    if (!m_thunderSystem) {
+        WTFLogAlways("CDMPrivateThunder: No active OpenCDM system for %s. Returning default robustness only.", m_keySystem.utf8().data());
+        return robustnesses;
+    }
+
+    char** buffer = nullptr;
+    uint16_t count = 0;
+
+    if (opencdm_system_supported_robustness(m_thunderSystem.get(), &buffer, &count) == ERROR_NONE && buffer != nullptr && count > 0) {
+        robustnesses.reserveCapacity(robustnesses.size() + count);
+        for (uint16_t i = 0; i < count; ++i) {
+            if (buffer[i] != nullptr) {
+                robustnesses.append(AtomString { String::fromUTF8(buffer[i]) });
+                free(buffer[i]);
+	    }
+        }
+    }
+    if (buffer != nullptr) {
+        free(buffer);
+    }
+    
+    return robustnesses;
+#endif
+
     return { emptyAtom(), "SW_SECURE_DECODE"_s, "SW_SECURE_CRYPTO"_s, "HW_SECURE_DECODE"_s, "HW_SECURE_CRYPTO"_s, "HW_SECURE_ALL"_s };
 }
 
-- 
2.44.4

