From 86bf1b0b5215e2ab5966fc9bac4f42918ec9de21 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Enrique=20Oca=C3=B1a=20Gonz=C3=A1lez?= <eocanha@igalia.com>
Date: Wed, 25 Feb 2026 11:21:14 +0100
Subject: [PATCH 1/2] Persistent licenses

---
 .../graphics/gstreamer/eme/CDMThunder.cpp     | 59 +++++++++++++++++--
 1 file changed, 55 insertions(+), 4 deletions(-)

diff --git a/Source/WebCore/platform/graphics/gstreamer/eme/CDMThunder.cpp b/Source/WebCore/platform/graphics/gstreamer/eme/CDMThunder.cpp
index 191303629b63..1a613d00bc2b 100644
--- a/Source/WebCore/platform/graphics/gstreamer/eme/CDMThunder.cpp
+++ b/Source/WebCore/platform/graphics/gstreamer/eme/CDMThunder.cpp
@@ -593,17 +593,65 @@ void CDMInstanceSessionThunder::updateLicense(const String& sessionID, LicenseTy
         sessionChanged(SessionChangedResult::Failure);
 }
 
-void CDMInstanceSessionThunder::loadSession(LicenseType, const String& sessionID, const String&, LoadSessionCallback&& callback)
+void CDMInstanceSessionThunder::loadSession(LicenseType licenseType, const String& sessionID, const String& /* origin */, LoadSessionCallback&& callback)
 {
-    ASSERT_UNUSED(sessionID, sessionID == m_sessionID);
+    ASSERT(isMainThread());
+
+    // MediaKeySession::load() is guaranteeing this (so we don't have to double-check):
+    // - MediaKeySession is not uninitialized (is initialized) and is not closed.
+    // - sessionID is sanitized.
+    // - LicenseType sessionType is not temporary (it's either PersistentUsageRecord or PersistentLicense).
+    // TODO: MediaKeySession::load() isn't checking if there are other open session with the same sessionId
+    //       because it lacks a global MediaKeySession tracker.
+
+    auto instance = cdmInstanceThunder();
+    ASSERT(instance);
+
+    GST_TRACE("Going to load session for session id %s", sessionID.utf8().data());
+
+    OpenCDMSession* session = nullptr;
+    OpenCDMError result = opencdm_construct_session(&instance->thunderSystem(), thunderLicenseType(licenseType), "",
+        nullptr, 0, sessionID.utf8().dataAsUInt8Ptr(), sessionID.length(), &m_thunderSessionCallbacks, this, &session);
+
+    if (!session) {
+        SessionLoadFailure failureReason { SessionLoadFailure::Other };
+        switch (result) {
+            case ERROR_NONE:
+                failureReason = SessionLoadFailure::None;
+                break;
+            case ERROR_KEYSYSTEM_NOT_SUPPORTED:
+            case ERROR_OUT_OF_MEMORY:
+            case ERROR_BUSY_CANNOT_INITIALIZE:
+            case ERROR_BUFFER_TOO_SMALL:
+                failureReason = SessionLoadFailure::QuotaExceeded;
+                break;
+            case ERROR_INVALID_SESSION:
+                failureReason = SessionLoadFailure::NoSessionData;            
+            default:
+                break;
+        }
 
+        GST_ERROR("Could not create session. OpenCDMError: %" PRIu32, static_cast<uint32_t>(result));
+
+        callback(std::nullopt, std::nullopt, std::nullopt, SuccessValue::Failed, failureReason);
+        return;
+    }
+
+    m_session = adoptInBoxPtr(session);
+    m_sessionID = String::fromUTF8(opencdm_session_id(m_session->get()));
+
+    GST_TRACE("Session created with id %s", m_sessionID.utf8().data());
+
+    ASSERT(sessionId == m_sessionID);
+
+    // FIXME: Is this callback for the response actually needed, since we're not doing generateRequest()?
     m_sessionChangedCallbacks.append([this, callback = WTFMove(callback)](bool success, RefPtr<SharedBuffer>&& responseMessage) mutable {
         ASSERT(isMainThread());
         if (success) {
             if (!responseMessage)
                 callback(m_keyStore.convertToJSKeyStatusVector(), std::nullopt, std::nullopt, SuccessValue::Succeeded, SessionLoadFailure::None);
             else {
-                // FIXME: Using JSON reponse messages is much cleaner than using string prefixes, I believe there
+                // FIXME: Using JSON response messages is much cleaner than using string prefixes, I believe there
                 // will even be other parts of the spec where not having structured data will be bad.
                 ParsedResponseMessage parsedResponseMessage(responseMessage);
                 ASSERT(parsedResponseMessage);
@@ -633,8 +681,11 @@ void CDMInstanceSessionThunder::loadSession(LicenseType, const String& sessionID
             }
         }
     });
-    if (!m_session || m_sessionID.isEmpty() || opencdm_session_load(m_session->get()))
+    result = OpenCDMError::ERROR_NONE;
+    if (!m_session || m_sessionID.isEmpty() || (result = opencdm_session_load(m_session->get()))) {
+        GST_DEBUG("loading failed. OpenCDMError: %" PRIu32, static_cast<uint32_t>(result));
         sessionChanged(SessionChangedResult::Failure);
+    }
 }
 
 void CDMInstanceSessionThunder::closeSession(const String& sessionID, CloseSessionCallback&& callback)
-- 
2.39.5 (Apple Git-154)

From 93aaf912e80c8f51c745d70dd01215450db2355c Mon Sep 17 00:00:00 2001
From: Eugene Mutavchi <Ievgen_Mutavchi@comcast.com>
Date: Mon, 16 Mar 2026 13:02:28 -0400
Subject: [PATCH 2/2] [EME][Thunder] fix complilation error and a crash when
 session cannot load

---
 Source/WebCore/platform/graphics/gstreamer/eme/CDMThunder.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Source/WebCore/platform/graphics/gstreamer/eme/CDMThunder.cpp b/Source/WebCore/platform/graphics/gstreamer/eme/CDMThunder.cpp
index 1a613d00bc2b..e05e31cd4983 100644
--- a/Source/WebCore/platform/graphics/gstreamer/eme/CDMThunder.cpp
+++ b/Source/WebCore/platform/graphics/gstreamer/eme/CDMThunder.cpp
@@ -611,7 +611,7 @@ void CDMInstanceSessionThunder::loadSession(LicenseType licenseType, const Strin
 
     OpenCDMSession* session = nullptr;
     OpenCDMError result = opencdm_construct_session(&instance->thunderSystem(), thunderLicenseType(licenseType), "",
-        nullptr, 0, sessionID.utf8().dataAsUInt8Ptr(), sessionID.length(), &m_thunderSessionCallbacks, this, &session);
+        nullptr, 0, reinterpret_cast<const uint8_t*>(sessionID.utf8().data()), sessionID.length(), &m_thunderSessionCallbacks, this, &session);
 
     if (!session) {
         SessionLoadFailure failureReason { SessionLoadFailure::Other };
@@ -671,7 +671,7 @@ void CDMInstanceSessionThunder::loadSession(LicenseType licenseType, const Strin
             }
         } else {
             GST_ERROR("session %s not loaded", m_sessionID.utf8().data());
-            if (responseMessage->isEmpty())
+            if (!responseMessage || responseMessage->isEmpty())
                 callback(std::nullopt, std::nullopt, std::nullopt, SuccessValue::Failed, sessionLoadFailureFromThunder({ }));
             else {
                 auto responseData = responseMessage->extractData();
-- 
2.39.5 (Apple Git-154)

