##########################################
Date: Tus, 15 Feb 2022 09:16:03 +0100
From: Ia0d327e697cfc205a5b07feb7a510ff075da29a3 Tus, 15 Feb 2022 09:16:03 +0100
Subject: [PATCH] Start Hdcp service after Hdmi service
Upstream-Status: Pending
Signed-off-by: Christine Tai <chrsitinetai@realtek.com>

#########################################
diff --git a/conf/hdcp.service b/conf/hdcp.service
index 44c40d6..799c289 100755
--- a/conf/hdcp.service
+++ b/conf/hdcp.service
@@ -18,7 +18,7 @@
 ##########################################################################
 [Unit]
 Description=HDCP Service
-After=dsmgr.service iarmbusd.service
+After=dsmgr.service iarmbusd.service hdmiservice.service
 
 [Service]
 ExecStart=/bin/sh -c '/usr/bin/tenableHDCP true >& /var/log/hdcp.log '
