From a8b02873075ada37d0dbe3ac8e5b9b7810b9a74f Mon Sep 17 00:00:00 2001
From: lsampa662 <lavanya_sampangi@comcast.com>
Source: Comcast
Upstream-Status: NA
Signed-off-by: lsampa662 <lavanya_sampangi@comcast.com>
Date: Tue, 25 May 2021 17:18:55 +0000
Subject: [PATCH] 0001-add_gstcaps_forcobalt_mediatype.patch

---
 plugin/CapsParser.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plugin/CapsParser.cpp b/plugin/CapsParser.cpp
index 7d77cff..4ebdd3d 100644
--- a/plugin/CapsParser.cpp
+++ b/plugin/CapsParser.cpp
@@ -111,12 +111,16 @@ namespace WPEFramework {
                 if(_lastHash != info_hash) {
                     LOG(eTrace, "Got a new info string %s hash = %ld\n", infoStr.c_str(), info_hash);
                     _lastHash = info_hash;
+                     ::string substring =infoStr.substr(0,5);
 
                     // Parse the data
                     ::string result = FindMarker(infoStr, _mediaTag);
                     if(!result.empty()) {
                         SetMediaType(result);
                     }
+                    else if(substring.compare("audio")==0||substring.compare("video")== 0){
+                        SetMediaType(substring);
+                    }
                     else {
                         LOG(eError, "No result for media type\n");
                     }
-- 
2.24.0

