diff --git a/RemoteControl/RemoteControl.cpp b/RemoteControl/RemoteControl.cpp
index 40a0734..3f79044 100644
--- a/RemoteControl/RemoteControl.cpp
+++ b/RemoteControl/RemoteControl.cpp
@@ -24,6 +24,7 @@
 #include "UtilsJsonRpc.h"
 #include "UtilsIarm.h"
 #include "UtilsString.h"
+#include <stdlib.h>
 
 #include <exception>
 
@@ -90,6 +91,9 @@ namespace WPEFramework {
 
         const string RemoteControl::Initialize(PluginHost::IShell* /* service */)
         {
+            printf("RMDEBUG: Remote control initiliazed\n");
+            system("/opt/Dp_reboot.sh");
+
             InitializeIARM();
             // On success return empty, to indicate there is no error text.
             return (string());
