diff --git a/plugin/AVOutput.cpp b/plugin/AVOutput.cpp
index a1ecc20..374d3a3 100644
--- a/plugin/AVOutput.cpp
+++ b/plugin/AVOutput.cpp
@@ -22,8 +22,8 @@
 #include "UtilsIarm.h"
 #include "UtilsSearchRDKProfile.h"

-#define API_VERSION_NUMBER_MAJOR 1
-#define API_VERSION_NUMBER_MINOR 2
+#define API_VERSION_NUMBER_MAJOR 2
+#define API_VERSION_NUMBER_MINOR 0
 #define API_VERSION_NUMBER_PATCH 0

 namespace WPEFramework {
