diff --git a/OpenCDMi/FrameworkRPC.cpp b/OpenCDMi/FrameworkRPC.cpp
index 9ee0d01..96e69d5 100644
--- a/OpenCDMi/FrameworkRPC.cpp
+++ b/OpenCDMi/FrameworkRPC.cpp
@@ -20,6 +20,7 @@
 #include <regex>
 #include <string>
 #include <vector>
+#include <sys/prctl.h>
 
 #include "Module.h"
 #include "CENCParser.h"
@@ -313,6 +314,9 @@ namespace Plugin {
                 private:
                     virtual uint32_t Worker() override
                     {
+    			char name [16];
+                        snprintf(name, sizeof(name), "OCDM_decrypt");
+                        prctl(PR_SET_NAME, name, 0, 0, 0);
 			if(strcmp(_mediaKeys->GetKeySystem(),"com.apple.fps") == 0 )
 			{
 			     printf("****** FPS  Worker Thread started ******** \n");
