From 6b384758e1e59f7cfd822ba835f64b0b0682f411 Mon Sep 17 00:00:00 2001
From: Krishna Priya Kanagaraj <krishnapriya_kanagaraj@comcast.com>
Date: Wed, 13 May 2026 08:05:45 +0000
Subject: [PATCH] add implementation of api to get robustnesslevel

---
 plugin/FrameworkRPC.cpp | 57 ++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 56 insertions(+), 1 deletion(-)

diff --git a/plugin/FrameworkRPC.cpp b/plugin/FrameworkRPC.cpp
index c17bdea..b4df7de 100644
--- a/plugin/FrameworkRPC.cpp
+++ b/plugin/FrameworkRPC.cpp
@@ -804,6 +804,40 @@ namespace Plugin {
                 END_INTERFACE_MAP
             };
 
+            class WidevineRobustnessExtensionProxy : public Exchange::IWidevineRobustnessExtension {
+            private:
+                CDMi::IWidevineRobustnessExtension* _impl;
+            public:
+                explicit WidevineRobustnessExtensionProxy(CDMi::IWidevineRobustnessExtension* impl)
+                    : _impl(impl) {
+                }
+
+                static Exchange::IWidevineRobustnessExtension* Create(CDMi::IWidevineRobustnessExtension* impl) {
+                    return Core::Service<WidevineRobustnessExtensionProxy>::Create<Exchange::IWidevineRobustnessExtension>(impl);
+                }
+
+                Exchange::OCDM_RESULT GetSupportedRobustness(RPC::IStringIterator*& robustness) const override {
+                    
+                    Exchange::OCDM_RESULT result = Exchange::OCDM_RESULT::OCDM_INTERFACE_NOT_IMPLEMENTED;
+
+                    if (!_impl)
+                        return result;
+
+                    std::list<std::string> levels;
+                    TRACE(Trace::Information, ("Get Supported Robustness()"));
+                    result = static_cast<Exchange::OCDM_RESULT>(_impl->GetSupportedRobustness(levels));
+
+		    if (result == Exchange::OCDM_RESULT::OCDM_SUCCESS && !levels.empty()) {
+                        robustness = Core::Service<RPC::StringIterator>::Create<RPC::IStringIterator>(levels);
+                    }
+                    return result;
+                }
+
+                BEGIN_INTERFACE_MAP(WidevineRobustnessExtensionProxy)
+                INTERFACE_ENTRY(Exchange::IWidevineRobustnessExtension)
+                END_INTERFACE_MAP
+            };
+
         public:
             AccessorOCDM(OCDMImplementation* parent, const string& name, const uint32_t defaultSize)
                 : _parent(*parent)
@@ -812,6 +846,7 @@ namespace Plugin {
                 , _defaultSize(defaultSize)
                 , _sessionList()
                 , _googleCastAuthExtension(GoogleCastAuthExtensionProxy::Create(parent->GoogleCastAuthExtension()))
+                , _widevineRobustnessExtension(WidevineRobustnessExtensionProxy::Create(parent->WidevineRobustnessExtension()))
             {
                 ASSERT(parent != nullptr);
             }
@@ -821,6 +856,11 @@ namespace Plugin {
                     _googleCastAuthExtension->Release();
                     _googleCastAuthExtension = nullptr;
                 }
+
+                if (_widevineRobustnessExtension) {
+                    _widevineRobustnessExtension->Release();
+                    _widevineRobustnessExtension = nullptr;
+                }
                 TRACE(Trace::Information, (_T("Released the AccessorOCDM server side [%d]"), __LINE__));
             }
 
@@ -1090,7 +1130,8 @@ namespace Plugin {
             BEGIN_INTERFACE_MAP(AccessorOCDM)
             INTERFACE_ENTRY(Exchange::IAccessorOCDM)
             INTERFACE_AGGREGATE(Exchange::IGoogleCastAuthExtension, _googleCastAuthExtension)
-            END_INTERFACE_MAP
+            INTERFACE_AGGREGATE(Exchange::IWidevineRobustnessExtension, _widevineRobustnessExtension)
+	    END_INTERFACE_MAP
 
         private:
             Exchange::ISession* FindSession(const CommonEncryptionData& keyIds, const string& keySystem) const
@@ -1162,6 +1203,7 @@ namespace Plugin {
             uint32_t _defaultSize;
             std::list<SessionImplementation*> _sessionList;
             Exchange::IGoogleCastAuthExtension* _googleCastAuthExtension;
+            Exchange::IWidevineRobustnessExtension*  _widevineRobustnessExtension;
         };
 
         class Config : public Core::JSON::Container {
@@ -1565,6 +1607,19 @@ namespace Plugin {
             return result;
         }
 
+        CDMi::IWidevineRobustnessExtension* WidevineRobustnessExtension()
+        {
+            CDMi::IWidevineRobustnessExtension* result = nullptr;
+            std::map<const std::string, SystemFactory>::const_iterator index(_systemToFactory.begin());
+            while (index != _systemToFactory.end()) {
+                auto instance = index->second.Factory->Instance();
+                if ((result = dynamic_cast<CDMi::IWidevineRobustnessExtension*>(instance)) != nullptr)
+                    break;
+                index++;
+            }
+            return result;
+        }
+
     private:
         void LoadDesignators(const string& keySystem, std::list<string>& designators) const
         {
-- 
2.44.4

