Date: Jun 24, 2020 6:09 PM
From: 85b53581887aa485147600711a9570d6e0e0bfe8 Jun 24, 2020 6:09 PM
Subject: [PATCH] RDK-28320: Integrate thunder packager plugin

     Build packager plugin

Source: COMCAST
Upstream-Status: Pending
Signed-off-by: Deva <Deva_Thiyagarajan2@comcast.com>
---
Index: git/Packager/opkg.conf.in
===================================================================
--- git.orig/Packager/opkg.conf.in
+++ git/Packager/opkg.conf.in
@@ -1,6 +1,6 @@
-src/gz snapshots http://yoctobuild.metrological.com/snapshots/vodafone
-src/gz local file:///hdd/cache/packages
-dest root /
-dest ram /tmp
-lists_dir ext /var/opkg-lists
-option overlay_root /overlay
+dest root /tmp
+option lists_dir /tmp/opkg/lists
+option lock_file /run/opkg/opkg.lock
+option connect_timeout_ms 30000
+option transfer_timeout_ms 600000
+option follow_location 1
Index: git/Packager/PackagerImplementation.cpp
===================================================================
--- git.orig/Packager/PackagerImplementation.cpp
+++ git/Packager/PackagerImplementation.cpp
@@ -37,7 +37,7 @@ namespace Plugin {
         opkg_config->conf_file = strdup(_configFile.c_str());
         opkg_config->tmp_dir = strdup(_tempPath.c_str());
         opkg_config->host_cache_dir = 1;
-        opkg_config->cache_dir = strdup(_cachePath.c_str());
+        opkg_config->cache_dir = strdup("/media/apps");
         opkg_config->verbosity = _verbosity;
         opkg_config->nodeps = _noDeps;
 
