From: Karunakaran A <Karunakaran_Amirthalingam@cable.comcast.com>
Date: 19 May 2023 17:24:43 +0000
Subject: [PATCH] rdkservice changes for R4 compilation
---
Index: git/plugin/FrameworkRPC.cpp
===================================================================
--- git.orig/plugin/FrameworkRPC.cpp
+++ git/plugin/FrameworkRPC.cpp
@@ -665,8 +665,12 @@ namespace Plugin {
                     TRACE(Trace::Information, (_T("ResetOutputProtection! %p"), this));
                     _mediaKeySession->ResetOutputProtection();
                 }
-
-                virtual void SetParameter(const std::string& name, const std::string& value) override {
+#ifdef USE_THUNDER_R4
+                virtual void SetParameter(const std::string& name, const std::string& value)
+#else
+                virtual void SetParameter(const std::string& name, const std::string& value) override
+#endif /* USE_THUNDER_R4 */
+                {
                     TRACE(Trace::Information, (_T("SetParameter! %p"), this));
                     _mediaKeySession->SetParameter(name, value);
                 }
