diff --git a/AVInput/AVInput.cpp b/AVInput/AVInput.cpp
index c758670..f4c8960 100644
--- a/AVInput/AVInput.cpp
+++ b/AVInput/AVInput.cpp
@@ -543,12 +543,13 @@ bool AVInput::setVideoRectangle(int x, int y, int width, int height, int type)
 uint32_t AVInput::getInputDevicesWrapper(const JsonObject& parameters, JsonObject& response)
 {
     LOGINFOMETHOD();
-
+    fprintf(stderr, "bvanav-dbg: JSONRPC request org.rdk.AVInput.1.getInputDevices entered plugin's getInputDevicesWrapper\n");
     if (parameters.HasLabel("typeOfInput")) {
         string sType = parameters["typeOfInput"].String();
         int iType = 0;
         try {
             iType = getTypeOfInput (sType);
+            fprintf(stderr, "bvanav-dbg: getInputDevicesWrapper - after 1st getTypeOfInput\n");
         }catch (...) {
             LOGWARN("Invalid Arguments");
             returnResponse(false);
@@ -558,11 +559,13 @@ uint32_t AVInput::getInputDevicesWrapper(const JsonObject& parameters, JsonObjec
     else {
         JsonArray listHdmi = getInputDevices(HDMI);
         JsonArray listComposite = getInputDevices(COMPOSITE);
+        fprintf(stderr, "bvanav-dbg: getInputDevicesWrapper - after 2nd getInputDevices\n");
         for (int i = 0; i < listComposite.Length(); i++) {
             listHdmi.Add(listComposite.Get(i));
         }
         response["devices"] = listHdmi;
     }
+    fprintf(stderr, "bvanav-dbg: JSONRPC request org.rdk.AVInput.1.getInputDevices returned from plugin's getInputDevicesWrapper\n");
     returnResponse(true);
 }
 
@@ -612,6 +615,7 @@ uint32_t AVInput::readEDIDWrapper(const JsonObject& parameters, JsonObject& resp
 
 JsonArray AVInput::getInputDevices(int iType)
 {
+    fprintf(stderr, "bvanav-dbg: org.rdk.AVInput.1.getInputDevices entered AVInput::getInputDevices iType: %d\n", iType);
     JsonArray list;
     try
     {
@@ -622,6 +626,7 @@ JsonArray AVInput::getInputDevices(int iType)
         else if (iType == COMPOSITE) {
             num = device::CompositeInput::getInstance().getNumberOfInputs();
         }
+        fprintf(stderr, "bvanav-dbg: AVInput::getInputDevices iType: %d, after device::HdmiInput::getInstance().getNumberOfInputs\n", iType);
         if (num > 0) {
             int i = 0;
             for (i = 0; i < num; i++) {
@@ -646,6 +651,7 @@ JsonArray AVInput::getInputDevices(int iType)
     catch (const std::exception &e) {
         LOGWARN("AVInputService::getInputDevices Failed");
     }
+    fprintf(stderr, "bvanav-dbg: org.rdk.AVInput.1.getInputDevices returned from AVInput::getInputDevices\n");
     return list;
 }
 
