diff --git a/PowerManager/PowerManagerImplementation.cpp b/PowerManager/PowerManagerImplementation.cpp
index e83df62..6722c63 100644
--- a/PowerManager/PowerManagerImplementation.cpp
+++ b/PowerManager/PowerManagerImplementation.cpp
@@ -17,6 +17,12 @@
  * limitations under the License.
  */
 
+#if defined (USE_WAKEUP_TIMER_EVT) && defined (_DISABLE_SCHD_REBOOT_AT_DEEPSLEEP) && defined (PLATCO_BOOTTO_STANDBY) && defined (MFR_TEMP_CLOCK_READ)
+#CFlags_enabled
+#else
+#CFlags_disabled
+#endif
+
 #include "PowerManagerImplementation.h"
 
 #include "UtilsIarm.h"
