diff --git a/Source/core/SystemInfo.h b/Source/core/SystemInfo.h
index 8da350dda..18222a7f4 100644
--- a/Source/core/SystemInfo.h
+++ b/Source/core/SystemInfo.h
@@ -87,6 +87,11 @@ namespace Core {
             return m_totalram;
         }

+        inline uint64_t GetMemUnit() const
+        {
+            return m_memunit;
+        }
+
         inline uint32_t GetPageSize() const
         {
            return m_pageSize;
@@ -214,6 +219,7 @@ namespace Core {
         const string m_HostName;

         uint64_t m_totalram;
+        uint64_t m_memunit{1};
         uint32_t m_pageSize;
         mutable uint32_t m_uptime;
         mutable uint64_t m_freeram;
diff --git a/Source/core/SystemInfo.cpp b/Source/core/SystemInfo.cpp
index de6705fee..a9cd650f8 100644
--- a/Source/core/SystemInfo.cpp
+++ b/Source/core/SystemInfo.cpp
@@ -152,6 +152,7 @@ namespace Core {
 #ifdef __APPLE__
         m_uptime = 0;
         m_totalram = 0;
+        m_memunit = info.mem_unit;
         m_pageSize = 0;
         m_freeram = 0;
         m_totalswap=0;
@@ -169,6 +170,7 @@ namespace Core {

         m_uptime = info.uptime;
         m_totalram = info.totalram;
+        m_memunit = info.mem_unit;
         m_pageSize = static_cast<uint32_t>(sysconf(_SC_PAGESIZE));
         m_freeram = info.freeram;
         m_totalswap = info.totalswap;
