diff --git a/Source/WPEFramework/PluginServer.h b/Source/WPEFramework/PluginServer.h
index 643c8450d..c3540ad62 100644
--- a/Source/WPEFramework/PluginServer.h
+++ b/Source/WPEFramework/PluginServer.h
@@ -3624,8 +3624,9 @@ POP_WARNING()
                     ASSERT(_service.IsValid() == true);
                     return _service->Callsign();
                 }
-            private:
+            protected: // bvanav-dbg
                 uint32_t _ID;
+            private:
                 Server* _server;
                 Core::ProxyType<Service> _service;
             };
@@ -3681,6 +3682,8 @@ POP_WARNING()
                             }
 
                             if (message->IsSet()) {
+                                // bvanav-dbg
+                                Core::JsonRpcProfiler::Instance().logInvoke(Job::_ID); 
                                 Core::ProxyType<Core::JSONRPC::Message> body = Job::Process(_token, message);
 
                                 // If we have no response body, it looks like an async-call...
@@ -3726,6 +3729,8 @@ POP_WARNING()
                         if (response->CacheControl.IsSet() == false)
                             response->CacheControl = _T("no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0");
 
+                        // bvanav-dbg
+                        Core::JsonRpcProfiler::Instance().logResponse(Job::_ID); 
                         Job::Submit(response);
 
                         if (_request->Connection.Value() == Web::Request::CONNECTION_CLOSE) {
@@ -4126,8 +4131,12 @@ POP_WARNING()
 
                         if (job.IsValid() == true) {
                             Core::ProxyType<Web::Request> baseRequest(request);
+                            // bvanav-dbg
+                            Core::ProxyType<Core::JSONRPC::Message> message(baseRequest->Body<Core::JSONRPC::Message>());
+                            Core::JsonRpcProfiler::Instance().logRequest(Id(), std::string(message->Designator.Value()));
                             job->Set(Id(), &_parent, service, baseRequest, _security->Token(), !request->ServiceCall());
-                            _parent.Submit(Core::ProxyType<Core::IDispatch>(job));
+                            _parent.Submit(Core::ProxyType<Core::IDispatch>(job), Id());
+                            // _parent.Submit(Core::ProxyType<Core::IDispatch>(job));
                         }
                     }
                     break;
@@ -4572,6 +4581,12 @@ POP_WARNING()
         {
             _dispatcher.Submit(job);
         }
+        // bvanav-dbg
+        inline void Submit(const Core::ProxyType<Core::IDispatch>& job, uint32_t channelId)
+        {
+            Core::JsonRpcProfiler::Instance().logQueue(channelId, _dispatcher.queueLength());
+            Submit(job);
+        }
         inline void Schedule(const uint64_t time, const Core::ProxyType<Core::IDispatch>& job)
         {
             _dispatcher.Schedule(time, job);
diff --git a/Source/core/CMakeLists.txt b/Source/core/CMakeLists.txt
index a8ab81fa0..4832a2bac 100644
--- a/Source/core/CMakeLists.txt
+++ b/Source/core/CMakeLists.txt
@@ -55,6 +55,7 @@ add_library(${TARGET} SHARED
         WorkerPool.cpp
         XGetopt.cpp
         ResourceMonitor.cpp
+        Profiler.cpp
         )
 
 
@@ -144,6 +145,7 @@ set(PUBLIC_HEADERS
         CallsignTLS.h
         TokenizedStringList.h
         MessageStore.h
+        Profiler.h
         )
 
 target_compile_options (${TARGET} PRIVATE -Wno-psabi)
diff --git a/Source/core/Profiler.cpp b/Source/core/Profiler.cpp
index e69de29bb..092548975 100644
--- a/Source/core/Profiler.cpp
+++ b/Source/core/Profiler.cpp
@@ -0,0 +1,119 @@
+#include "Profiler.h"
+
+namespace WPEFramework {
+
+namespace Core {
+
+    /* static */ JsonRpcProfiler& JsonRpcProfiler::Instance()
+    {
+        static JsonRpcProfiler& instance = SingletonType<JsonRpcProfiler>::Instance();
+        return (instance);
+    }
+
+    JsonRpcProfiler::JsonRpcProfiler()
+    {
+        std::ifstream logFile(_fileName);
+        bool fileExists = logFile.good();
+        logFile.close();
+
+        _logFile.open(_fileName, std::ios::app);
+
+        if(!fileExists)
+        {
+            std::cout << "JsonRpcProfiler [INFO]: Writing to file " << _fileName << std::endl;
+
+            _logFile << "PluginApi" << ", RequestType" << ", QueueLength" << ", RequestTimeStamp" 
+                    << ", InvokeTimeStamp"  << ", ThunderLatency(ms)" << ", ResponseTimeStamp" 
+                    << ", PluginLatency(ms)" << ", TotalLatency(ms)" << std::endl;
+        }
+        std::cout << "JsonRpcProfiler [INFO]: Init done !!" << std::endl;
+    }
+    
+    JsonRpcProfiler::~JsonRpcProfiler() 
+    {
+        _logFile.close();
+        std::cout << "JsonRpcProfiler [INFO]: DeInit done !!" << std::endl;
+    }
+
+    std::string JsonRpcProfiler::timeStamp(const timePoint& tp)
+    {
+        std::time_t tt = std::chrono::system_clock::to_time_t(tp);
+        std::tm tm;
+        gmtime_r(&tt, &tm);
+        auto ms = std::chrono::duration_cast<std::chrono::milliseconds>(tp.time_since_epoch()) % 1000;
+        char buffer[40];
+        std::snprintf(buffer, sizeof(buffer),
+            "%04d-%02d-%02dT%02d:%02d:%02d.%03ldZ",
+            tm.tm_year + 1900,
+            tm.tm_mon + 1,
+            tm.tm_mday,
+            tm.tm_hour,
+            tm.tm_min,
+            tm.tm_sec,
+            static_cast<long>(ms.count()));
+        return std::string(buffer);
+    }
+
+    std::chrono::milliseconds::rep JsonRpcProfiler::elapsedTime(const timePoint& tp1, const timePoint& tp2)
+    {
+        return std::chrono::duration_cast<std::chrono::milliseconds>(tp1 - tp2).count();
+    }
+
+    void JsonRpcProfiler::log(uint32_t channelId)
+    {
+        _logFile << _map[channelId].api << ", " << _map[channelId].type 
+                << ", " << _map[channelId].queueLength 
+                << ", " << timeStamp(_map[channelId].requestTime) 
+                << ", " << timeStamp(_map[channelId].invokeTime)
+                << ", " << elapsedTime(_map[channelId].invokeTime, _map[channelId].requestTime)
+                << ", " << timeStamp(_map[channelId].responseTime)
+                << ", " << elapsedTime(_map[channelId].responseTime, _map[channelId].invokeTime)
+                << ", " << elapsedTime(_map[channelId].responseTime, _map[channelId].requestTime) << std::endl;  
+    }
+
+    void JsonRpcProfiler::logRequest(uint32_t channelId, const std::string& api)
+    {
+        if(_map.find(channelId) == _map.end())
+        {
+            _map.emplace(channelId, Info());
+            _map[channelId].api = api;
+            _map[channelId].requestTime = std::chrono::system_clock::now();
+            _map[channelId].type = "JSON-RPC";
+        }
+        else
+            std::cout << "JsonRpcProfiler [ERROR][logRequest]: Duplicate Channel Id "  << channelId << " for " << api << std::endl;
+    }
+
+    void JsonRpcProfiler::logQueue(uint32_t channelId, uint32_t len)
+    {
+        if(_map.find(channelId) != _map.end())
+        {
+            _map[channelId].queueLength = len;
+        }
+        else
+            std::cout << "JsonRpcProfiler [ERROR][logQueue]: Channel Id "  << channelId << " not found !!" << std::endl;
+    }
+
+    void JsonRpcProfiler::logInvoke(uint32_t channelId)
+    {
+        if(_map.find(channelId) != _map.end())
+        {
+            _map[channelId].invokeTime = std::chrono::system_clock::now();
+        }
+        else
+            std::cout << "JsonRpcProfiler [ERROR][logQueue]: Channel Id "  << channelId << " not found !!" << std::endl;
+    }
+
+    void JsonRpcProfiler::logResponse(uint32_t channelId)
+    {
+        if(_map.find(channelId) != _map.end())
+        {
+            _map[channelId].responseTime = std::chrono::system_clock::now();
+            log(channelId);
+            _map.erase(channelId);
+        }
+        else
+            std::cout << "JsonRpcProfiler [ERROR][logResponse]: Channel Id "  << channelId << " not found !!" << std::endl;
+    }
+}
+} 
\ No newline at end of file
diff --git a/Source/core/Profiler.h b/Source/core/Profiler.h
index e69de29bb..fed722b92 100644
--- a/Source/core/Profiler.h
+++ b/Source/core/Profiler.h
@@ -0,0 +1,64 @@
+// bvanav-dbg
+#ifndef PROFILER_H
+#define PROFILER_H
+
+#include "Module.h"
+#include "Portability.h"
+#include "Singleton.h"
+#include <chrono>
+#include <ctime>
+#include <fstream>
+#include <iostream>
+#include <unordered_map>
+namespace WPEFramework {
+
+namespace Core {
+
+    class EXTERNAL JsonRpcProfiler {
+        typedef std::chrono::system_clock::time_point timePoint;
+
+        private:
+            JsonRpcProfiler();
+            JsonRpcProfiler(const JsonRpcProfiler&) = delete;
+            JsonRpcProfiler& operator=(const JsonRpcProfiler&) = delete;
+            friend class SingletonType<JsonRpcProfiler>;
+
+            struct Info
+            {
+                Info()
+                    : api(), queueLength(0), requestTime(), invokeTime(), responseTime()
+                {}
+
+                std::string api;
+                std::string type;
+                uint32_t queueLength;
+                timePoint requestTime;
+                timePoint invokeTime;
+                timePoint responseTime;
+            };
+
+            static constexpr const char* _fileName = "/opt/logs/thunder-api-latency.csv";
+
+            /* helpers */
+            std::chrono::milliseconds::rep elapsedTime(const timePoint& tp1, const timePoint& tp2);
+            string timeStamp(const timePoint& tp);
+            void log(uint32_t channelId);
+
+        public:
+            static JsonRpcProfiler& Instance();
+            ~JsonRpcProfiler();
+
+            void logRequest(uint32_t channelId, const string& api);
+            void logQueue(uint32_t channelId, uint32_t len);
+            void logInvoke(uint32_t channelId);
+            void logResponse(uint32_t channelId);
+
+        private: 
+            std::ofstream _logFile;
+            std::unordered_map<int, Info> _map;
+    };
+
+}
+} 
+
+#endif
\ No newline at end of file
diff --git a/Source/core/WorkerPool.h b/Source/core/WorkerPool.h
index ea68bc1a8..29e0cf359 100644
--- a/Source/core/WorkerPool.h
+++ b/Source/core/WorkerPool.h
@@ -420,7 +420,12 @@ POP_WARNING()
             #endif
             _threadPool.Stop();
         }
-
+        // bvanav-dbg
+        inline uint32_t queueLength()
+        {
+            return _threadPool.Pending();
+        }
+        
     private:
         Scheduler _scheduler;
         ThreadPool _threadPool;
diff --git a/Source/core/core.h b/Source/core/core.h
index 42749c1d8..a03e80fc7 100644
--- a/Source/core/core.h
+++ b/Source/core/core.h
@@ -103,6 +103,7 @@
 #include "CallsignTLS.h"
 #include "TokenizedStringList.h"
 #include "MessageStore.h"
+#include "Profiler.h" // bvanav-dbg
 
 #ifdef __WINDOWS__
 #pragma comment(lib, "core.lib")
