diff --git a/Source/WPEFramework/PluginHost.cpp b/Source/WPEFramework/PluginHost.cpp
index fcafedc38..8a26ecc58 100644
--- a/Source/WPEFramework/PluginHost.cpp
+++ b/Source/WPEFramework/PluginHost.cpp
@@ -699,7 +699,9 @@ POP_WARNING()
             // We need at least the loopback interface before we continue...
             StartLoopbackInterface();
 #endif
-
+            // bvanv-dbg
+            SYSLOG(Logging::JsonRpcProfiler, (string(_T("Fromat: Api, QueueLength, ThunderLatency(ns), PluginLatency(ns), TotalLatency(ns)"))));
+            
             // Startup/load/initialize what we found in the configuration.
             _dispatcher = new PluginHost::Server(*_config, _background);
 
diff --git a/Source/WPEFramework/PluginServer.h b/Source/WPEFramework/PluginServer.h
index 643c8450d..a4dd6cc86 100644
--- a/Source/WPEFramework/PluginServer.h
+++ b/Source/WPEFramework/PluginServer.h
@@ -3638,6 +3638,7 @@ POP_WARNING()
                     : Job()
                     , _request()
                     , _jsonrpc(false)
+                    , _outer() // bvanav-dbg
                 {
                 }
                 ~WebRequestJob() override
@@ -3655,7 +3656,7 @@ POP_WARNING()
                     _missingResponse->ErrorCode = Web::STATUS_INTERNAL_SERVER_ERROR;
                     _missingResponse->Message = _T("There is no response from the requested service.");
                 }
-                void Set(const uint32_t id, Server* server, Core::ProxyType<Service>& service, Core::ProxyType<Web::Request>& request, const string& token, const bool JSONRPC)
+                void Set(const uint32_t id, Server* server, Core::ProxyType<Service>& service, Core::ProxyType<Web::Request>& request, const string& token, const bool JSONRPC, Channel* outer) // bvanav-dbg
                 {
                     Job::Set(id, server, service);
 
@@ -3664,6 +3665,7 @@ POP_WARNING()
                     _request = request;
                     _jsonrpc = JSONRPC && (_request->HasBody() == true);
                     _token = token;
+                    _outer = outer; // bvanav-dbg
                 }
                 void Dispatch() override
                 {
@@ -3681,7 +3683,9 @@ POP_WARNING()
                             }
 
                             if (message->IsSet()) {
+                                _outer->setInvokeTime(START);  // bvanav-dbg
                                 Core::ProxyType<Core::JSONRPC::Message> body = Job::Process(_token, message);
+                                _outer->setInvokeTime(END); // bvanav-dbg
 
                                 // If we have no response body, it looks like an async-call...
                                 if (body.IsValid() == false) {
@@ -3756,6 +3760,7 @@ POP_WARNING()
                 Core::ProxyType<Web::Request> _request;
                 string _token;
                 bool _jsonrpc;
+                Channel* _outer; // bvanav-dbg
 
                 static Core::ProxyType<Web::Response> _missingResponse;
             };
@@ -4126,7 +4131,14 @@ POP_WARNING()
 
                         if (job.IsValid() == true) {
                             Core::ProxyType<Web::Request> baseRequest(request);
-                            job->Set(Id(), &_parent, service, baseRequest, _security->Token(), !request->ServiceCall());
+                            job->Set(Id(), &_parent, service, baseRequest, _security->Token(), !request->ServiceCall(),this);
+                            // bvanav-dbg
+                            if (!request->ServiceCall() && (baseRequest->HasBody() == true))
+                            {
+                                Core::ProxyType<Core::JSONRPC::Message> message(baseRequest->Body<Core::JSONRPC::Message>());
+                                setJsonRpcMethod(message->Designator.Value());
+                                setQueueLength(_parent.WorkerPool().queueLength());
+                            }
                             _parent.Submit(Core::ProxyType<Core::IDispatch>(job));
                         }
                     }
@@ -4315,7 +4327,41 @@ POP_WARNING()
             {
                 SetId(id);
             }
+        // bvanav-dbg
+        public:
+            inline void setRequestTime()
+            {
+                requestTime = Core::Time::Now();
+            }
+            enum invokeTime {START, END};
+            inline void setInvokeTime(invokeTime time)
+            {
+                if(time == START)
+                    invokeTimeStart = Core::Time::Now();
+                else if(time == END)
+                    invokeTimeEnd = Core::Time::Now();
+            }
+            inline void setResponseTime()
+            {
+                responseTime = Core::Time::Now();
 
+                if(!jsonRpcMethod.empty())
+                {
+                    Core::Time::nanoseconds thunderLatency =  (invokeTimeStart.nanosec() - requestTime.nanosec()) 
+                                                                + (responseTime.nanosec() - invokeTimeEnd.nanosec());
+                    Core::Time::nanoseconds pluginLatency =  invokeTimeEnd.nanosec() - invokeTimeStart.nanosec();
+                    Core::Time::nanoseconds totalLatency = thunderLatency + pluginLatency;
+                    SYSLOG(Logging::JsonRpcProfiler, (_T("%s, %d, %ld, %ld, %ld"), jsonRpcMethod.c_str(), queueSize, thunderLatency, pluginLatency, totalLatency));
+                }
+            }
+            inline void setJsonRpcMethod(string method)
+            {
+                jsonRpcMethod = method;
+            }
+            inline void setQueueLength(uint32_t size)
+            {
+                queueSize = size; 
+            }
         private:
             inline string SelectSupportedProtocol(const Web::ProtocolsArray& protocols)
             {
@@ -4347,6 +4393,14 @@ POP_WARNING()
             Core::ProxyType<Service> _service;
             bool _requestClose;
 
+            // bvanav-dbg
+            Core::Time requestTime;
+            Core::Time invokeTimeStart;
+            Core::Time invokeTimeEnd;
+            Core::Time responseTime;
+            string jsonRpcMethod;
+            uint32_t queueSize;
+
             // Factories for creating jobs that can be placed on the PluginHost Worker pool.
             static Core::ProxyPoolType<WebRequestJob> _webJobs;
             static Core::ProxyPoolType<JSONElementJob> _jsonJobs;
diff --git a/Source/core/SocketPort.cpp b/Source/core/SocketPort.cpp
index 0898183df..3afee5104 100644
--- a/Source/core/SocketPort.cpp
+++ b/Source/core/SocketPort.cpp
@@ -1110,7 +1110,8 @@ namespace WPEFramework {
             bool dataLeftToSend = true;
 
             m_syncAdmin.Lock();
-
+            setResponseTime(); // bvanav-dbg
+            
             m_State &= (~(SocketPort::WRITE | SocketPort::WRITESLOT));
 
             while (((m_State & (SocketPort::WRITE | SocketPort::SHUTDOWN | SocketPort::OPEN | SocketPort::EXCEPTION)) == SocketPort::OPEN) && (dataLeftToSend == true)) {
@@ -1216,6 +1217,7 @@ namespace WPEFramework {
                 }
 
                 if (m_ReadBytes != 0) {
+                    setRequestTime(); // bvanav-dbg
                     uint16_t handledBytes = ReceiveData(m_ReceiveBuffer, m_ReadBytes);
 
                     ASSERT(m_ReadBytes >= handledBytes);
diff --git a/Source/core/SocketPort.h b/Source/core/SocketPort.h
index 103816293..3b3635a5e 100644
--- a/Source/core/SocketPort.h
+++ b/Source/core/SocketPort.h
@@ -259,6 +259,15 @@ namespace WPEFramework {
             virtual uint32_t Initialize();
             virtual int32_t Read(uint8_t buffer[], const uint16_t length) const;
             virtual int32_t Write(const uint8_t buffer[], const uint16_t length);
+            // bvanav-dbg
+            virtual void setRequestTime()
+            {
+                ASSERT(false);
+            }
+            virtual void setResponseTime()
+            {
+                ASSERT(false);
+            }
             void SetError() {
                 m_State |= SocketPort::EXCEPTION;
             }
@@ -461,6 +470,16 @@ namespace WPEFramework {
                         _parent.Accept(newClient, remoteId);
                     }
                 }
+            // bvanav-dbg
+            protected:
+                virtual void setRequestTime()
+                {
+                    ASSERT(false);
+                }
+                virtual void setResponseTime()
+                {
+                    ASSERT(false);
+                }
 
             private:
                 SocketListner& _parent;
diff --git a/Source/core/Time.cpp b/Source/core/Time.cpp
index 97a59c4ae..6acfde98d 100644
--- a/Source/core/Time.cpp
+++ b/Source/core/Time.cpp
@@ -1025,6 +1025,12 @@ POP_WARNING()
         return ((static_cast<microsecondsfromepoch>(_time.tv_sec) * MicroSecondsPerSecond ) + (static_cast<microsecondsfromepoch>(_time.tv_nsec)/NanoSecondsPerMicroSecond));
     }
 
+    // bvanav-dbg
+    Time::nanoseconds Time::nanosec() const
+    {
+        return ((static_cast<nanoseconds>(_time.tv_sec) * NanoSecondsPerSecond ) + static_cast<nanoseconds>(_time.tv_nsec));
+    }
+
     uint8_t Time::DayOfWeek() const
     {
         return DayOfWeek(TMHandle());
diff --git a/Source/core/Time.h b/Source/core/Time.h
index 3ef93a5d9..ace394c79 100644
--- a/Source/core/Time.h
+++ b/Source/core/Time.h
@@ -51,6 +51,8 @@ namespace Core {
         static constexpr uint32_t NanoSecondsPerMicroSecond = 1000;
         static constexpr uint32_t NanoSecondsPerMilliSecond = MicroSecondsPerMilliSecond * MicroSecondsPerMilliSecond;
         static constexpr uint32_t MicroSecondsPerSecond = MilliSecondsPerSecond * MicroSecondsPerMilliSecond;
+        // bvanav-dbg
+        static constexpr uint32_t NanoSecondsPerSecond = MilliSecondsPerSecond * NanoSecondsPerMilliSecond;
 
         // Difference in Seconds between UNIX and NTP epoch (25567).
         static constexpr uint32_t SecondsPerMinute = 60;
@@ -63,7 +65,7 @@ namespace Core {
 
         // relative to 1st of January 1970, 00:00:00 in microseconds
         using microsecondsfromepoch = uint64_t;
-
+        using nanoseconds = uint64_t; // bvanav-dbg
 #ifdef __POSIX__
 
 public:
@@ -229,6 +231,8 @@ public:
 
         // UTC time
         microsecondsfromepoch Ticks() const;
+        // bvanav-dbg
+        nanoseconds nanosec() const;
 
         bool FromRFC1123(const string& buffer);
         bool FromRFC1036(const string& buffer);
diff --git a/Source/core/WorkerPool.h b/Source/core/WorkerPool.h
index ea68bc1a8..29e0cf359 100644
--- a/Source/core/WorkerPool.h
+++ b/Source/core/WorkerPool.h
@@ -420,7 +420,12 @@ POP_WARNING()
             #endif
             _threadPool.Stop();
         }
-
+        // bvanav-dbg
+        inline uint32_t queueLength()
+        {
+            return _threadPool.Pending();
+        }
+        
     private:
         Scheduler _scheduler;
         ThreadPool _threadPool;
diff --git a/Source/messaging/Logging.cpp b/Source/messaging/Logging.cpp
index 602c3f35f..6cf0cdb77 100644
--- a/Source/messaging/Logging.cpp
+++ b/Source/messaging/Logging.cpp
@@ -33,6 +33,7 @@ namespace Logging {
     SYSLOG_ANNOUNCE(Error);
     SYSLOG_ANNOUNCE(ParsingError);
     SYSLOG_ANNOUNCE(Notification);
+    SYSLOG_ANNOUNCE(JsonRpcProfiler); // bvanav-dbg
 
     static const TCHAR* UnknownCallsign = {_T("NoTLSCallsign") };
 
diff --git a/Source/messaging/LoggingCategories.h b/Source/messaging/LoggingCategories.h
index 2a1cf7397..7dd3d9749 100644
--- a/Source/messaging/LoggingCategories.h
+++ b/Source/messaging/LoggingCategories.h
@@ -34,6 +34,7 @@ namespace Logging {
     DEFINE_LOGGING_CATEGORY(ParsingError);
     DEFINE_LOGGING_CATEGORY(Fatal);
     DEFINE_LOGGING_CATEGORY(Crash);
+    DEFINE_LOGGING_CATEGORY(JsonRpcProfiler); // bvanav-dbg
 
 } // namespace Logging
 }
diff --git a/Source/websocket/WebSocketLink.h b/Source/websocket/WebSocketLink.h
index 4583bcc82..fb76b0ce9 100644
--- a/Source/websocket/WebSocketLink.h
+++ b/Source/websocket/WebSocketLink.h
@@ -710,6 +710,16 @@ POP_WARNING()
                 _adminLock.Unlock();
             }
 
+            // bvanav-dbg
+            void setRequestTime() override
+            {
+                _parent.setRequestTime();
+            }
+            void setResponseTime() override
+            {
+                _parent.setResponseTime();
+            }
+
         private:
             inline uint16_t State() const
             {
@@ -1101,6 +1111,13 @@ POP_WARNING()
         virtual bool IsIdle() const = 0;
 
     protected:
+        // bvanav-dbg
+        virtual void setRequestTime(){
+            ASSERT(false);
+        }
+        virtual void setResponseTime(){
+            ASSERT(false);
+        }
         void Lock() const {
             _channel.Lock();
         }
