diff --git a/Source/WPEFramework/PluginHost.cpp b/Source/WPEFramework/PluginHost.cpp
index 4b8d3d52a..7abce176f 100644
--- a/Source/WPEFramework/PluginHost.cpp
+++ b/Source/WPEFramework/PluginHost.cpp
@@ -785,6 +785,9 @@ POP_WARNING()
             // Startup/load/initialize what we found in the configuration.
             _dispatcher = new PluginHost::Server(*_config, _background);
 
+            SYSLOG(Logging::JsonRpcProfiler, (string(_T("ActivateRequest-Fromat: Type, ChannelId, RequestCount, SrcPort, QueueLength, Method, Params")))); // bvanav-dbg
+            SYSLOG(Logging::JsonRpcProfiler, (string(_T("PluginRequest-Fromat: Type, ChannelId, RequestCount, SrcPort, QueueLength, Method")))); // bvanav-dbg
+            SYSLOG(Logging::JsonRpcProfiler, (string(_T("Response-Fromat: Type, ChannelId, RequestCount, QueueLength, Method, ThunderLatency(ns), PluginLatency(ns), TotalLatency(ns)")))); // bvanav-dbg
             SYSLOG(Logging::Startup, (_T(EXPAND_AND_QUOTE(APPLICATION_NAME) " actively listening.")));
 
             // If we have handlers open up the gates to analyze...
diff --git a/Source/WPEFramework/PluginServer.h b/Source/WPEFramework/PluginServer.h
index b6a9d4604..b4e2424a5 100644
--- a/Source/WPEFramework/PluginServer.h
+++ b/Source/WPEFramework/PluginServer.h
@@ -27,6 +27,10 @@
 #include "WarningReportingCategories.h"
 #include "PostMortem.h"
 #include <atomic>
+#include <cinttypes> // bvanav-dbg
+#include <chrono> // bvanav-dbg
+#include <unordered_map> // bvanav-dbg
+#include <mutex> // bvanav-dbg
 
 #ifdef PROCESSCONTAINERS_ENABLED
 #include "../processcontainers/ProcessContainer.h"
@@ -3716,6 +3720,8 @@ POP_WARNING()
         // (is closed) during the service process, the ChannelMap will
         // not find it and just "flush" the presented work.
         class Channel : public PluginHost::Channel {
+            using Clock = std::chrono::steady_clock; // bvanav-dbg
+            enum RequestType {HTTP, WS}; // bvanav-dbg
         public:
             class Job : public Core::IDispatch {
             public:
@@ -3825,6 +3831,7 @@ POP_WARNING()
                     : Job()
                     , _request()
                     , _jsonrpc(false)
+                    , _outer() // bvanav-dbg
                 {
                 }
                 ~WebRequestJob() override
@@ -3842,7 +3849,7 @@ POP_WARNING()
                     _missingResponse->ErrorCode = Web::STATUS_INTERNAL_SERVER_ERROR;
                     _missingResponse->Message = _T("There is no response from the requested service.");
                 }
-                void Set(const uint32_t id, Server* server, Core::ProxyType<Service>& service, Core::ProxyType<Web::Request>& request, const string& token, const bool JSONRPC)
+                void Set(const uint32_t id, Server* server, Core::ProxyType<Service>& service, Core::ProxyType<Web::Request>& request, const string& token, const bool JSONRPC, Channel* outer, uint8_t requestCount) // bvanav-dbg
                 {
                     Job::Set(id, server, service);
 
@@ -3851,6 +3858,8 @@ POP_WARNING()
                     _request = request;
                     _jsonrpc = JSONRPC && (_request->HasBody() == true);
                     _token = token;
+                    _outer = outer; // bvanav-dbg
+                    _requestCount = requestCount; // bvanav-dbg
                 }
                 void Dispatch() override
                 {
@@ -3868,7 +3877,12 @@ POP_WARNING()
                             }
 
                             if (message->IsSet()) {
+                                // SYSLOG(Logging::JsonRpcProfiler, (_T("bvanavdbg HTTP %ld, %s, %d invoke-start"), _outer->Id(), _outer->_jsonRpcTracerMap[_requestCount].api.c_str(), _requestCount));
+                                Clock::time_point invokeStart = Clock::now(); // bvanav-dbg
                                 Core::ProxyType<Core::JSONRPC::Message> body = Job::Process(_token, message);
+                                Clock::time_point invokeEnd = Clock::now();  // bvanav-dbg
+                                // SYSLOG(Logging::JsonRpcProfiler, (_T("bvanavdbg HTTP %ld, %s, %d invoke-end"), _outer->Id(), _outer->_jsonRpcTracerMap[_requestCount].api.c_str(), _requestCount));
+                                _outer->setpluginLatency(static_cast<uint64_t>(std::chrono::duration_cast<std::chrono::nanoseconds>(invokeEnd - invokeStart).count()), _requestCount, RequestType::HTTP); // bvanav-dbg
 
                                 // If we have no response body, it looks like an async-call...
                                 if (body.IsValid() == false) {
@@ -3913,6 +3927,7 @@ POP_WARNING()
                         if (response->CacheControl.IsSet() == false)
                             response->CacheControl = _T("no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0");
 
+                        response->setRequestCount(_requestCount); // bvanav-dbg
                         Job::Submit(response);
 
                         if (_request->Connection.Value() == Web::Request::CONNECTION_CLOSE) {
@@ -3943,6 +3958,8 @@ POP_WARNING()
                 Core::ProxyType<Web::Request> _request;
                 string _token;
                 bool _jsonrpc;
+                Channel* _outer; // bvanav-dbg
+                uint8_t _requestCount; // bvanav-dbg
 
                 static Core::ProxyType<Web::Response> _missingResponse;
             };
@@ -3955,6 +3972,7 @@ POP_WARNING()
                     : Job()
                     , _element()
                     , _jsonrpc(false)
+                    , _outer() // bvanav-dbg
                 {
                 }
                 ~JSONElementJob() override
@@ -3967,7 +3985,7 @@ POP_WARNING()
                 }
 
             public:
-                void Set(const uint32_t id, Server* server, Core::ProxyType<Service>& service, Core::ProxyType<Core::JSON::IElement>& element, const string& token, const bool JSONRPC)
+                void Set(const uint32_t id, Server* server, Core::ProxyType<Service>& service, Core::ProxyType<Core::JSON::IElement>& element, const string& token, const bool JSONRPC, Channel* outer, uint8_t requestCount) // bvanav-dbg
                 {
                     Job::Set(id, server, service);
 
@@ -3976,6 +3994,8 @@ POP_WARNING()
                     _element = element;
                     _jsonrpc = JSONRPC;
                     _token = token;
+                    _outer = outer; // bvanav-dbg
+                    _requestCount = requestCount; // bvanav-dbg
                 }
                 void Dispatch() override
                 {
@@ -3993,9 +4013,12 @@ POP_WARNING()
                         if (HasService() == true) {
                             message->ImplicitCallsign(GetService().Callsign());
                         }
-
+                        // SYSLOG(Logging::JsonRpcProfiler, (_T("bvanavdbg WS %ld, %s, %d invoke-start"), _outer->Id(), _outer->_jsonRpcTracerMap[_requestCount].api.c_str(), _requestCount));
+                        Clock::time_point invokeStart = Clock::now(); // bvanav-dbg
                         _element = Core::ProxyType<Core::JSON::IElement>(Job::Process(_token, message));
-
+                        Clock::time_point invokeEnd = Clock::now();  // bvanav-dbg
+                        // SYSLOG(Logging::JsonRpcProfiler, (_T("bvanavdbg WS %ld, %s, %d invoke-end"), _outer->Id(), _outer->_jsonRpcTracerMap[_requestCount].api.c_str(), _requestCount));
+                        _outer->setpluginLatency(static_cast<uint64_t>(std::chrono::duration_cast<std::chrono::nanoseconds>(invokeEnd - invokeStart).count()), _requestCount, RequestType::WS); // bvanav-dbg
 #if THUNDER_PERFORMANCE
                         tracking->Execution();
 #endif
@@ -4006,6 +4029,7 @@ POP_WARNING()
 
                     if (_element.IsValid()) {
                         // Fire and forget, We are done !!!
+                        _element->setRequestCount(_requestCount); // bvanav-dbg
                         Job::Submit(_element);
                         _element.Release();
                     }
@@ -4028,6 +4052,8 @@ POP_WARNING()
                 Core::ProxyType<Core::JSON::IElement> _element;
                 string _token;
                 bool _jsonrpc;
+                Channel* _outer; // bvanav-dbg
+                uint8_t _requestCount; // bvanav-dbg
             };
             class TextJob : public Job {
             public:
@@ -4315,7 +4341,25 @@ POP_WARNING()
 
                         if (job.IsValid() == true) {
                             Core::ProxyType<Web::Request> baseRequest(request);
-                            job->Set(Id(), &_parent, service, baseRequest, _security->Token(), !request->ServiceCall());
+                            
+                            // bvanav-dbg
+                            uint8_t requestCount;
+                            if (!request->ServiceCall() && (baseRequest->HasBody() == true))
+                            {
+                                Core::ProxyType<Core::JSONRPC::Message> message(baseRequest->Body<Core::JSONRPC::Message>());
+                                uint32_t queueSize = _parent.WorkerPool().queueLength();
+                                requestCount = setJsonRpcMethod(message->Designator.Value(), queueSize);
+                                if(_jsonRpcTracerMap[requestCount].api == "Controller.1.activate" || _jsonRpcTracerMap[requestCount].api == "activate")
+                                {
+                                    SYSLOG(Logging::JsonRpcProfiler, (_T("ActivateRequest HTTP, %ld, %d, %ld, %ld, %s, %s"), Id(), requestCount, _remotePort, _jsonRpcTracerMap[requestCount].pendingQueue, _jsonRpcTracerMap[requestCount].api.c_str(), message->Parameters.Value().c_str()));
+                                }
+                                else
+                                {
+                                    SYSLOG(Logging::JsonRpcProfiler, (_T("PluginRequest HTTP, %ld, %d, %ld, %ld, %s"), Id(), requestCount, _remotePort, _jsonRpcTracerMap[requestCount].pendingQueue, _jsonRpcTracerMap[requestCount].api.c_str()));
+                                }
+                            }
+                            job->Set(Id(), &_parent, service, baseRequest, _security->Token(), !request->ServiceCall(), this, requestCount); // bvanav-dbg
+
                             _parent.Submit(Core::ProxyType<Core::IDispatch>(job));
                         }
                     }
@@ -4392,7 +4436,24 @@ POP_WARNING()
                     ASSERT(job.IsValid() == true);
 
                     if ((_service.IsValid() == true) && (job.IsValid() == true)) {
-                        job->Set(Id(), &_parent, _service, element, _security->Token(), ((State() & Channel::JSONRPC) != 0));
+                        
+                        // bvanav-dbg
+                        uint8_t requestCount;
+                        Core::ProxyType<Core::JSONRPC::Message> message(element);
+                        if (message.IsValid())
+                        {
+                            uint32_t queueSize = _parent.WorkerPool().queueLength();
+                            requestCount = setJsonRpcMethod(message->Designator.Value(), queueSize);
+                            if(_jsonRpcTracerMap[requestCount].api == "activate" || _jsonRpcTracerMap[requestCount].api == "Controller.1.activate")
+                            {
+                                SYSLOG(Logging::JsonRpcProfiler, (_T("ActivateRequest WS, %ld, %d, %ld, %ld, %s, %s"), Id(), requestCount, _remotePort, _jsonRpcTracerMap[requestCount].pendingQueue, _jsonRpcTracerMap[requestCount].api.c_str(), message->Parameters.Value().c_str()));
+                            }
+                            else
+                            {
+                                SYSLOG(Logging::JsonRpcProfiler, (_T("PluginRequest WS, %ld, %d, %ld, %ld, %s"), Id(), requestCount, _remotePort, _jsonRpcTracerMap[requestCount].pendingQueue, _jsonRpcTracerMap[requestCount].api.c_str()));
+                            }
+                        }
+                        job->Set(Id(), &_parent, _service, element, _security->Token(), ((State() & Channel::JSONRPC) != 0), this, requestCount); // bvanav-dbg
                         _parent.Submit(Core::ProxyType<Core::IDispatch>(job));
                     }
                 }
@@ -4512,6 +4573,43 @@ POP_WARNING()
                 }
             }
 
+        // bvanav-dbg
+        public:
+            inline void setRequestTime(uint16_t port = 0)
+            {
+                _remotePort = port;
+                _requestTime = Clock::now();
+            }
+            inline void setpluginLatency(uint64_t latency /*ns*/, uint8_t requestCount, RequestType type)
+            {
+                std::lock_guard<std::mutex> lock(_invokeMtx); 
+                _jsonRpcTracerMap[requestCount].pluginLatency = latency;
+                _jsonRpcTracerMap[requestCount].type = type;
+            }
+            inline void setResponseTime(uint8_t requestCount)
+            {
+                _responseTime = Clock::now();
+                
+                if(_jsonRpcTracerMap.find(requestCount) != _jsonRpcTracerMap.end() && _jsonRpcTracerMap[requestCount].pluginLatency > 0)
+                {
+                    _responseTime = Clock::now();
+                    _thunderLatency =  static_cast<uint64_t>(std::chrono::duration_cast<std::chrono::nanoseconds>(_responseTime - _jsonRpcTracerMap[requestCount].requestTime).count()) - _jsonRpcTracerMap[requestCount].pluginLatency;
+                    _totalLatency = _thunderLatency + _jsonRpcTracerMap[requestCount].pluginLatency;
+                    SYSLOG(Logging::JsonRpcProfiler, (_T("Response %s, %ld, %d, %ld, %s, %" PRIu64", %" PRIu64", %" PRIu64), (_jsonRpcTracerMap[requestCount].type == RequestType::HTTP ? "HTTP" : "WS"), Id(), requestCount, _jsonRpcTracerMap[requestCount].pendingQueue, _jsonRpcTracerMap[requestCount].api.c_str(), _thunderLatency, _jsonRpcTracerMap[requestCount].pluginLatency, _totalLatency));
+                    
+                    _jsonRpcTracerMap.erase(requestCount);
+                }
+            }
+            uint8_t setJsonRpcMethod(string method, uint32_t queueLen)
+            {
+                _requestCount++;
+                _jsonRpcTracerMap[_requestCount].api = method;
+                _jsonRpcTracerMap[_requestCount].pendingQueue = queueLen; 
+                _jsonRpcTracerMap[_requestCount].requestTime = _requestTime;
+                _jsonRpcTracerMap[_requestCount].pluginLatency = 0;
+                return _requestCount;
+            }
+
         private:
             inline string SelectSupportedProtocol(const Web::ProtocolsArray& protocols)
             {
@@ -4544,6 +4642,22 @@ POP_WARNING()
             bool _requestClose;
             std::atomic<bool> _serviceCleanedUp;
 
+        // bvanav-dbg
+        public:
+            Clock::time_point _requestTime;
+            Clock::time_point _responseTime;
+            Clock::time_point _invokeFinishTime;
+            string _jsonRpcMethod;
+            uint64_t _thunderLatency;
+            uint64_t _totalLatency;
+            uint32_t _queueSize;
+            uint16_t _remotePort;
+            uint8_t _requestCount;
+            struct TraceInfo {RequestType type; string api; uint32_t pendingQueue; Clock::time_point requestTime; uint64_t pluginLatency;};
+            std::unordered_map<uint8_t, TraceInfo> _jsonRpcTracerMap;
+            std::mutex _invokeMtx;
+        
+        private: //bvanav-dbg
             // Factories for creating jobs that can be placed on the PluginHost Worker pool.
             static Core::ProxyPoolType<WebRequestJob> _webJobs;
             static Core::ProxyPoolType<JSONElementJob> _jsonJobs;
diff --git a/Source/core/JSON.h b/Source/core/JSON.h
index 7d14be8a1..d7466cd9c 100644
--- a/Source/core/JSON.h
+++ b/Source/core/JSON.h
@@ -307,6 +307,21 @@ namespace Core {
                 return loaded;
             }
             virtual uint16_t Deserialize(const char stream[], const uint16_t maxLength, uint32_t& offset, Core::OptionalType<Error>& error) = 0;
+        
+        // bvanav-dbg
+        public:
+            inline void setRequestCount(uint8_t num)
+            {
+                _requestCount = num;
+            }
+            inline uint8_t getRequestCount()
+            {
+                return _requestCount;
+            }
+
+        private:
+            uint8_t _requestCount;
+        
         };
 
         struct EXTERNAL IMessagePack {
diff --git a/Source/core/SocketPort.cpp b/Source/core/SocketPort.cpp
index 5eb722290..d9ed6a476 100644
--- a/Source/core/SocketPort.cpp
+++ b/Source/core/SocketPort.cpp
@@ -1122,6 +1122,7 @@ namespace WPEFramework {
             bool dataLeftToSend = true;
 
             m_syncAdmin.Lock();
+            setResponseTime(); // bvanav-dbg
 
             m_State &= (~(SocketPort::WRITE | SocketPort::WRITESLOT));
 
@@ -1230,6 +1231,7 @@ namespace WPEFramework {
                 }
 
                 if (m_ReadBytes != 0) {
+                    setRequestTime(RemoteNode().PortNumber()); // bvanav-dbg
                     uint16_t handledBytes = ReceiveData(m_ReceiveBuffer, m_ReadBytes);
 
                     ASSERT(m_ReadBytes >= handledBytes);
diff --git a/Source/core/SocketPort.h b/Source/core/SocketPort.h
index f21dded93..3585b95cc 100644
--- a/Source/core/SocketPort.h
+++ b/Source/core/SocketPort.h
@@ -261,6 +261,15 @@ namespace WPEFramework {
             virtual uint32_t Initialize();
             virtual int32_t Read(uint8_t buffer[], const uint16_t length) const;
             virtual int32_t Write(const uint8_t buffer[], const uint16_t length);
+            // bvanav-dbg
+            virtual void setRequestTime(uint16_t port = 0)
+            {
+                ASSERT(false);
+            }
+            virtual void setResponseTime()
+            {
+                ASSERT(false);
+            }
             void SetError() {
                 m_State |= SocketPort::EXCEPTION;
             }
@@ -463,7 +472,16 @@ namespace WPEFramework {
                         _parent.Accept(newClient, remoteId);
                     }
                 }
-
+            // bvanav-dbg
+            protected:
+                virtual void setRequestTime(uint16_t port = 0)
+                {
+                    ASSERT(false);
+                }
+                virtual void setResponseTime()
+                {
+                    ASSERT(false);
+                }
             private:
                 SocketListner& _parent;
             };
diff --git a/Source/core/WorkerPool.h b/Source/core/WorkerPool.h
index 83b29f245..ccc36e13b 100644
--- a/Source/core/WorkerPool.h
+++ b/Source/core/WorkerPool.h
@@ -423,7 +423,12 @@ POP_WARNING()
             #endif
             _threadPool.Stop();
         }
-
+        // bvanav-dbg
+        inline uint32_t queueLength()
+        {
+            return _threadPool.Pending();
+        }
+        
     private:
         Scheduler _scheduler;
         ThreadPool _threadPool;
diff --git a/Source/messaging/Logging.cpp b/Source/messaging/Logging.cpp
index 7fdd08890..545c7bc2a 100644
--- a/Source/messaging/Logging.cpp
+++ b/Source/messaging/Logging.cpp
@@ -42,6 +42,7 @@ namespace Logging {
                 SYSLOG_ANNOUNCE(Error)
                 SYSLOG_ANNOUNCE(ParsingError)
                 SYSLOG_ANNOUNCE(Notification)
+                SYSLOG_ANNOUNCE(JsonRpcProfiler); // bvanav-dbg
             }
         } ControlsRegistration;
 
diff --git a/Source/messaging/LoggingCategories.h b/Source/messaging/LoggingCategories.h
index 38584f199..58c734459 100644
--- a/Source/messaging/LoggingCategories.h
+++ b/Source/messaging/LoggingCategories.h
@@ -33,6 +33,7 @@ namespace Logging {
     DEFINE_LOGGING_CATEGORY(ParsingError)
     DEFINE_LOGGING_CATEGORY(Fatal)
     DEFINE_LOGGING_CATEGORY(Crash)
+    DEFINE_LOGGING_CATEGORY(JsonRpcProfiler); // bvanav-dbg
 
 } // namespace Logging
 }
diff --git a/Source/plugins/Channel.h b/Source/plugins/Channel.h
index 7ab9cf0db..18feeb7c4 100644
--- a/Source/plugins/Channel.h
+++ b/Source/plugins/Channel.h
@@ -288,7 +288,7 @@ namespace PluginHost {
                 BaseClass::Unlock();
 
                 if (trigger == true) {
-                    BaseClass::Trigger();
+                    BaseClass::Trigger(entry); // bvanav-dbg
                 }
             }
         }
diff --git a/Source/websocket/WebResponse.h b/Source/websocket/WebResponse.h
index aa51ea776..7207f0abb 100644
--- a/Source/websocket/WebResponse.h
+++ b/Source/websocket/WebResponse.h
@@ -474,10 +474,18 @@ POP_WARNING()
         {
             return (_marshalMode);
         }
-
+        inline void setRequestCount(uint8_t num) // bvanav-dbg
+        {
+            _requestCount = num;
+        }
+        inline uint8_t getRequestCount() // bvanav-dbg
+        {
+            return _requestCount;
+        }
     private:
         Core::ProxyType<IBody> _body;
         MarshalType _marshalMode;
+        uint8_t _requestCount; // bvanav-dbg
     };
 }
 }
diff --git a/Source/websocket/WebSocketLink.h b/Source/websocket/WebSocketLink.h
index 0d4bb1898..90ad02f83 100644
--- a/Source/websocket/WebSocketLink.h
+++ b/Source/websocket/WebSocketLink.h
@@ -508,7 +508,7 @@ POP_WARNING()
             void Submit(const Core::ProxyType<OUTBOUND>& element)
             {
                 _adminLock.Lock();
-
+                _requestCount = element->getRequestCount(); // bvanav-dbg
                 if ((IsSuspended() == false) && (IsOpen() == true) && (_serializerImpl.Submit(element) == true)) {
 
                     _adminLock.Unlock();
@@ -518,6 +518,14 @@ POP_WARNING()
                     _adminLock.Unlock();
                 }
             }
+            using ACTUALLINK::Trigger; // bvanav-dbg
+            void Trigger(const Core::ProxyType<Core::JSON::IElement>& element) // bvanav-dbg
+            {
+                // _adminLock.Lock();
+                _requestCount = element->getRequestCount();
+                // _adminLock.Unlock();
+                ACTUALLINK::Trigger();
+            }
             uint32_t Close(const uint32_t waitTime)
             {
                 uint32_t result = 0;
@@ -714,6 +722,16 @@ POP_WARNING()
                 _adminLock.Unlock();
             }
 
+            // bvanav-dbg
+            void setRequestTime(uint16_t port = 0) override
+            {
+                _parent.setRequestTime(port);
+            }
+            void setResponseTime() override
+            {
+                _parent.setResponseTime(_requestCount);
+            }
+
         private:
             inline uint16_t State() const
             {
@@ -932,6 +950,7 @@ POP_WARNING()
             string _commandData;
             Core::ProxyType<typename OUTBOUND::BaseElement> _webSocketMessage;
             uint64_t _pingFireTime;
+            uint8_t _requestCount; // bvanav-dbg
         };
 
     public:
@@ -1083,6 +1102,10 @@ POP_WARNING()
         {
             _channel.Ping();
         }
+        void Trigger(const Core::ProxyType<Core::JSON::IElement>& entry) // bvanav-dbg
+        {
+            _channel.Trigger(entry);
+        }
         void Trigger()
         {
             _channel.Trigger();
@@ -1105,6 +1128,13 @@ POP_WARNING()
         virtual bool IsIdle() const = 0;
 
     protected:
+        // bvanav-dbg
+        virtual void setRequestTime(uint16_t port = 0){
+            ASSERT(false);
+        }
+        virtual void setResponseTime(uint8_t requestCount){
+            ASSERT(false);
+        }
         void Lock() const {
             _channel.Lock();
         }
