diff --git a/Source/WPEFramework/PluginServer.cpp b/Source/WPEFramework/PluginServer.cpp
index 26609995..7cf05559 100644
--- a/Source/WPEFramework/PluginServer.cpp
+++ b/Source/WPEFramework/PluginServer.cpp
@@ -1137,7 +1137,7 @@ POP_WARNING()
         {
             if (service->State() != PluginHost::Service::state::UNAVAILABLE) {
                 if (service->Startup() == PluginHost::IShell::startup::ACTIVATED) {
-                    SYSLOG(Logging::Startup, (_T("Activating plugin [%s]:[%s]"),
+                    SYSLOG(Logging::Startup, (_T("Activating plugin [%s]:[%s], autostart is true"),
                         service->ClassName().c_str(), service->Callsign().c_str()));
                     service->Activate(PluginHost::IShell::STARTUP);
                 }
