diff --git a/Source/WPEFramework/Controller.cpp b/Source/WPEFramework/Controller.cpp
index 15cc523cd..88f3af798 100644
--- a/Source/WPEFramework/Controller.cpp
+++ b/Source/WPEFramework/Controller.cpp
@@ -877,6 +877,7 @@ namespace Plugin {

             if (_pluginServer->Services().FromIdentifier(callsign, service) == Core::ERROR_NONE) {
                 ASSERT(service.IsValid());
+                SYSLOG(Logging::Startup, (_T("Activating plugin [%s] Requested"),callsign.c_str()));
                 result = service->Activate(PluginHost::IShell::REQUESTED);

                 // Normalise return code
@@ -891,6 +892,7 @@ namespace Plugin {
         else {
             result = Core::ERROR_PRIVILIGED_REQUEST;
         }
+        SYSLOG(Logging::Startup, (_T("Activating plugin [%s] returned %u"),callsign.c_str(), result));
         return result;
     }
