diff --git a/Source/core/ResourceMonitor.h b/Source/core/ResourceMonitor.h
index b5a8f096..89985448 100644
--- a/Source/core/ResourceMonitor.h
+++ b/Source/core/ResourceMonitor.h
@@ -210,6 +210,7 @@ namespace Core {
             // Make sure this entry is only registered once !!!
             if (std::find(_resourceList.begin(), _resourceList.end(), &resource) == _resourceList.end()) {
                 _resourceList.push_back(&resource);
+                TRACE_L3("Registering resource %p with descriptor %d at index %d\n", &resource, resource.Descriptor(), static_cast<int>(_resourceList.size()));
             }
 
             if (_resourceList.size() == 1) {
diff --git a/Source/core/SocketPort.cpp b/Source/core/SocketPort.cpp
index 0898183d..68ae1199 100644
--- a/Source/core/SocketPort.cpp
+++ b/Source/core/SocketPort.cpp
@@ -407,10 +407,16 @@ namespace WPEFramework {
         SocketPort::~SocketPort()
         {
             TRACE_L5("Destructor SocketPort <%p>", (this));
+            TRACE_L3("Destructor SocketPort <%p>", (this));
 
             // Make sure the socket is closed before you destruct. Otherwise
             // the virtuals might be called, which are destructed at this point !!!!
             ASSERT((m_Socket == INVALID_SOCKET) || (IsClosed()));
+            if(!((m_Socket == INVALID_SOCKET) || (IsClosed()))) {
+                TRACE_L3("Watchout!!! Socket is still open in destructor of SocketPort <%p> m_socket %d\n", (this), m_Socket);
+                int* ptr = nullptr;
+                *ptr = 10;
+            }
 
             if (m_Socket != INVALID_SOCKET) {
                 DestroySocket(m_Socket);
diff --git a/Source/websocket/JSONRPCLink.h b/Source/websocket/JSONRPCLink.h
index d7dc2f08..db544aa0 100644
--- a/Source/websocket/JSONRPCLink.h
+++ b/Source/websocket/JSONRPCLink.h
@@ -1442,10 +1442,12 @@ PUSH_WARNING(DISABLE_WARNING_THIS_IN_MEMBER_INITIALIZER_LIST)
 				: _connection(*this, remoteCallsign, localCallsign, query)
 				, _callsign(remoteCallsign)
 			{
+				TRACE_L1("Creating SmartLinkType for callsign [%s]", remoteCallsign.c_str());
 			}
 POP_WARNING()
 			~SmartLinkType()
 			{
+				TRACE_L1("Destroying SmartLinkType for callsign [%s]", _callsign.c_str());
 			}
 		public:
 			template <typename INBOUND, typename METHOD>
