--- a/Source/WPEFramework/WPEFramework.conf.in	2026-03-09 12:57:24.911107355 +0000
+++ b/Source/WPEFramework/WPEFramework.conf.in	2026-03-09 12:59:40.557388985 +0000
@@ -134,6 +134,18 @@
   
   messaging.add("logging", __logging)
   
+  __info_PIS= {
+       "category" : "Information",
+       "module" : "Plugin_PluginInitializerService",
+       "enabled" : True
+      }
+
+  __warn_PIS= {
+       "category" : "Warning",
+       "module" : "Plugin_PluginInitializerService",
+       "enabled" : True
+      }
+
   __info_LS= {
        "category" : "Information",
        "module" : "Plugin_LocationSync",
@@ -161,6 +173,8 @@
   __tracing_settings = []
   __tracing_settings.append(__fatal)
   __tracing_settings.append(__error)
+  __tracing_settings.append(__warn_PIS)
+  __tracing_settings.append(__info_PIS)
   __tracing_settings.append(__warn_LS)
   __tracing_settings.append(__info_LS)
   __tracing_settings.append(__warn_MES)
