diff --git a/Source/core/JSONRPC.h b/Source/core/JSONRPC.h
index 8d52c7485..19e037580 100644
--- a/Source/core/JSONRPC.h
+++ b/Source/core/JSONRPC.h
@@ -148,13 +148,17 @@

                             if( frameworkError <= 999 ) {
                                 Code = static_cast<int32_t>(frameworkError);
+                               printf("If-FrameworkError:%d and Code:%d\n",frameworkError,Code.Value());
                             } else {
                                 Code = -31000 - static_cast<int32_t>(frameworkError);
+                               printf("Else-FrameworkError:%d and Code:%d\n",frameworkError,Code.Value());
                             }
                         } else {
                             Code = static_cast<int32_t>(frameworkError & 0x7FFFFFFF) + 500;
+                           printf("Not a com-rpc error-FrameworkError:%d and Code:%d\n",frameworkError,Code.Value());
                         }
                        Text = Core::ErrorToString(frameworkError);
+                       printf("Text:%s,Framework_error:%d\n",Text.Value().c_str(),frameworkError);
                         break;
                     }
                 }

diff --git a/Source/plugins/JSONRPC.h b/Source/plugins/JSONRPC.h
index 8f23df367..1ae2e22d4 100644
--- a/Source/plugins/JSONRPC.h
+++ b/Source/plugins/JSONRPC.h
@@ -34,14 +34,17 @@
         uint32_t InvokeOnHandler(const Core::JSONRPC::Context& context, const string& method, const string& parameters, string& response, Core::JSONRPC::Handler& handler, JSONRPCERRORASSESSORTYPE errorhandler)
         {
             uint32_t result = handler.Invoke(context, method, parameters, response);
+           printf("Plugins-JsonRPC-Result:%d\n",result);
             if(result != Core::ERROR_NONE) {
                 result = errorhandler(context, method, parameters, result, response);
+               printf("Plugins-JsonRPC-Result is not ERROR_NONE:%d\n",result);
             }
             return result;
         }
         template<>
         uint32_t InvokeOnHandler<void*>(const Core::JSONRPC::Context& context, const string& method, const string& parameters, string& response, Core::JSONRPC::Handler& handler, void*)
         {
+           printf("DBG-InvokeOnHandler");
             return handler.Invoke(context, method, parameters, response);
         }
     }
