From: Sergey Borushevsky
Date: May 20 22:53:26 2024
Subject: [PATCH] wpeframework sd notify patch

Upstream-Status: pending
Signed-off-by: Sergey Borushevsky <Sergey_Borushevsky2@cable.comcast.com>
---
Index: git/Source/WPEFramework/PluginServer.cpp
===================================================================
--- git.orig/Source/WPEFramework/PluginServer.cpp
+++ git/Source/WPEFramework/PluginServer.cpp
@@ -24,6 +24,10 @@
 #include <syslog.h>
 #endif
 
+#ifdef SYSTEMD_FOUND
+#include <systemd/sd-daemon.h>
+#endif /* SYSTEMD_FOUND */
+
 #ifdef PROCESSCONTAINERS_ENABLED
 #include "../processcontainers/ProcessContainer.h"
 #endif
@@ -980,6 +984,10 @@ POP_WARNING()
                 }
             }
         }
+#ifdef SYSTEMD_FOUND
+        SYSLOG(Logging::Startup, (_T("Notify that WPEFramework Systemd Service is Ready")));
+        sd_notify(0, "READY=1");
+#endif /* SYSTEMD_FOUND */
     }
 
     void Server::Close()
