diff --git a/Utils/PluginActivator/CMakeLists.txt b/Utils/PluginActivator/CMakeLists.txt
index bd465e3a..fd653138 100644
--- a/Utils/PluginActivator/CMakeLists.txt
+++ b/Utils/PluginActivator/CMakeLists.txt
@@ -36,6 +36,15 @@ target_link_libraries(${PROJECT_NAME}
     ${NAMESPACE}Plugins::${NAMESPACE}Plugins
 )

+find_package(Systemd)
+if (SYSTEMD_FOUND)
+    target_link_libraries(${PROJECT_NAME}
+            PRIVATE
+            Systemd::Systemd
+            )
+    SET(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -DSYSTEMD_FOUND")
+endif()
+
 install(
     TARGETS ${PROJECT_NAME}
     RUNTIME DESTINATION bin
diff --git a/Utils/PluginActivator/source/COMRPCStarter.cpp b/Utils/PluginActivator/source/COMRPCStarter.cpp
index f2d27ed4..4107ccdc 100644
--- a/Utils/PluginActivator/source/COMRPCStarter.cpp
+++ b/Utils/PluginActivator/source/COMRPCStarter.cpp
@@ -23,6 +23,7 @@

 #include <chrono>
 #include <thread>
+#include <systemd/sd-daemon.h>

 COMRPCStarter::COMRPCStarter(const string& pluginName)
     : IPluginStarter()
@@ -92,6 +93,8 @@ bool COMRPCStarter::activatePlugin(const uint8_t maxRetries, const uint16_t retr
             } else {
                 // Our work here is done!
                 LOG_INF(_pluginName.c_str(), "Successfully activated plugin after %dms", duration.MilliSeconds());
+                LOG_INF(_pluginName.c_str(), "Plugin is now activated sending READY notification");
+                sd_notify(0, "READY=1");
                 success = true;
             }
             lifetime->Release();
