diff --git a/Source/WPEFramework/CMakeLists.txt b/Source/WPEFramework/CMakeLists.txt 
index 13fe9d59d..17ba9459f 100644
--- a/Source/WPEFramework/CMakeLists.txt 
+++ b/Source/WPEFramework/CMakeLists.txt 
@@ -33,6 +33,9 @@
         PostMortem.cpp
         )

+separate_arguments(BREAKPAD_LDFLAGS)
+target_link_libraries(WPEFramework PRIVATE "${BREAKPAD_LDFLAGS}")
+
 target_compile_definitions(${TARGET}
         PRIVATE
           NAMESPACE=${NAMESPACE}
@@ -74,6 +77,8 @@
           ${RFC_LIBRARIES}
           Threads::Threads
         )
+
+

 if (PROCESSCONTAINERS)
     target_link_libraries(${TARGET}

