diff --git a/Source/core/Services.cpp b/Source/core/Services.cpp
index 393f0475..31ef57ce 100644
--- a/Source/core/Services.cpp
+++ b/Source/core/Services.cpp
@@ -103,7 +103,7 @@ namespace Core {
     void ServiceAdministrator::FlushLibraries()
     {
         _adminLock.Lock();
-        while (_unreferencedLibraries.size() != 0) {
+        if (_unreferencedLibraries.size() != 0) {
             // A few closing code instructions might still be required for 
             // that thread that submitted the librray to complete, so at 
             // least give that thread a slice to complete the last few 
@@ -111,6 +111,7 @@ namespace Core {
             // the last reference)
             std::this_thread::yield();
 
+            printf("Releasing library: %s\n", _unreferencedLibraries.front().Name().c_str());
             _unreferencedLibraries.pop_front();
         }
         _adminLock.Unlock();
diff --git a/Source/core/SocketPort.cpp b/Source/core/SocketPort.cpp
index 0898183d..6cbfccb5 100644
--- a/Source/core/SocketPort.cpp
+++ b/Source/core/SocketPort.cpp
@@ -411,8 +411,16 @@ namespace WPEFramework {
             // Make sure the socket is closed before you destruct. Otherwise
             // the virtuals might be called, which are destructed at this point !!!!
             ASSERT((m_Socket == INVALID_SOCKET) || (IsClosed()));
+            if (m_Socket != INVALID_SOCKET ) {
+                printf("[%s:%d-%s] Socket(%d) is not set to INVALID but reached dtor\n", __FILE__, __LINE__, __FUNCTION__, m_Socket );
+            }
+
+            if (IsClosed() == false) {
+                printf("[%s:%d-%s] Socket(%d) is NOT CLOSED but reached dtor\n", __FILE__, __LINE__, __FUNCTION__, m_Socket );
+            }
 
             if (m_Socket != INVALID_SOCKET) {
+                printf("[%s:%d-%s] Destroying Socket(%d) in dtor \n", __FILE__, __LINE__, __FUNCTION__, m_Socket );
                 DestroySocket(m_Socket);
             }
 
@@ -1250,6 +1258,8 @@ namespace WPEFramework {
 
             if (m_State != 0) {
                 result = false;
+                printf("In Closed m_State(%X) is not = 0\n", m_State.load());
+                // m_State = 0;
             }
             else {
                 DestroySocket(m_Socket);
