diff --git a/Source/com/Administrator.h b/Source/com/Administrator.h
index 5e9a488c..dcceb165 100644
--- a/Source/com/Administrator.h
+++ b/Source/com/Administrator.h
@@ -36,7 +36,7 @@ namespace RPC {
 #ifdef __DEBUG__
     enum { CommunicationTimeOut = Core::infinite }; // Time in ms. Forever
 #else
-    enum { CommunicationTimeOut = 20000 }; // Time in ms. 20 Seconds
+    enum { CommunicationTimeOut = Core::infinite }; // Time in ms. Forever
 #endif
     enum { CommunicationBufferSize = 8120 }; // 8K :-)

