diff --git a/Source/WPEProcess/Process.cpp b/Source/WPEProcess/Process.cpp
index 213da58d..cc1c7ee4 100644
--- a/Source/WPEProcess/Process.cpp
+++ b/Source/WPEProcess/Process.cpp
@@ -598,6 +598,7 @@
         WARNING_REPORTING_THREAD_SETCALLSIGN_GUARD(callsign.c_str());

         #ifdef USE_BREAKPAD
+       printf("USE_BREAKPAD_ENABLED:PostMortemPath-%s",PostMortemPath);
         google_breakpad::MinidumpDescriptor descriptor(options.PostMortemPath);
         google_breakpad::ExceptionHandler eh(descriptor, NULL,
             [](const google_breakpad::MinidumpDescriptor&, void*, bool succeeded)
index 213da58d..cc1c7ee4 100644
--- a/Source/WPEFramework/Config.h
+++ b/Source/WPEFramework/Config.h
@@ -362,7 +362,7 @@
                 , VolatilePath(_T("/tmp"))
 #endif
                 , ProxyStubPath()
-                , PostMortemPath(_T("/opt/minidumps"))
+                , PostMortemPath(_T("/opt/secure/minidumps"))
 #ifdef __WINDOWS__
                 , Communicator(_T("127.0.0.1:7889"))
 #else
