diff --git a/Source/com/Administrator.h b/Source/com/Administrator.h
index 99417a28..c48c8812 100644
--- a/Source/com/Administrator.h
+++ b/Source/com/Administrator.h
@@ -40,7 +40,7 @@ namespace RPC {
 #ifdef __DEBUG__
     enum { CommunicationTimeOut = Core::infinite }; // Time in ms. Forever
 #else
-    enum { CommunicationTimeOut = COMRPC_TIMEOUT }; // Time in ms.
+    enum { CommunicationTimeOut = Core::infinite }; // Time in ms.
 #endif
     enum { CommunicationBufferSize = 8120 }; // 8K :-)
