diff --git a/Source/WPEFramework/PluginServer.cpp b/Source/WPEFramework/PluginServer.cpp
index fa723305..22f071be 100644
--- a/Source/WPEFramework/PluginServer.cpp
+++ b/Source/WPEFramework/PluginServer.cpp
@@ -1093,11 +1093,13 @@ POP_WARNING()
         std::vector<PluginHost::ISubSystem::subsystem> externallyControlled;
         ServiceMap::Iterator iterator(_services.Services());
 
-#if 0
+#if 1
         // Load the metadata for the subsystem information..
         while (iterator.Next() == true)
         {
+            START_TIME("Loading Metadata", iterator->Callsign().c_str());
             iterator->LoadMetadata();
+            STOP_TIME_AND_PRINT_DURATION
             for (const PluginHost::ISubSystem::subsystem& entry : iterator->SubSystemControl()) {
                 Core::EnumerateType<PluginHost::ISubSystem::subsystem> name(entry);
                 if (std::find(externallyControlled.begin(), externallyControlled.end(), entry) != externallyControlled.end()) {
diff --git a/Source/WPEFramework/PluginServer.h b/Source/WPEFramework/PluginServer.h
index 643c8450..436376d4 100644
--- a/Source/WPEFramework/PluginServer.h
+++ b/Source/WPEFramework/PluginServer.h
@@ -1942,7 +1942,9 @@ namespace PluginHost {
                         ErrorMessage(_T("local class definitions/version does not exist"));
                     }
                 } else {
+                    START_TIME(__FUNCTION__, locator.c_str());
                     _library = LoadLibrary(locator);
+                    STOP_TIME_AND_PRINT_DURATION
                     if (_library.IsLoaded() == false) {
                         ErrorMessage(_T("Library could not be loaded"));
                     }
diff --git a/Source/core/Library.cpp b/Source/core/Library.cpp
index b38ffe78..302549fc 100644
--- a/Source/core/Library.cpp
+++ b/Source/core/Library.cpp
@@ -101,7 +101,9 @@ namespace Core {
         , _error()
     {
 #ifdef __LINUX__
+        START_TIME("dlopen", fileName);
         void* handle = dlopen(fileName, RTLD_LAZY);
+        STOP_TIME_AND_PRINT_DURATION
 #endif
 #ifdef __WINDOWS__
         HMODULE handle = ::LoadLibrary(fileName);
diff --git a/Source/core/Portability.h b/Source/core/Portability.h
index c63b0e3d..ff32c9a1 100644
--- a/Source/core/Portability.h
+++ b/Source/core/Portability.h
@@ -965,4 +965,24 @@ namespace std {
 #define THUNDER_VERSION_MAJOR 4
 #define THUNDER_VERSION_MINOR 4
 
+
+#define START_TIME(FUNCTION_NAME, REMARKS)                                   \
+    auto __start_time = std::chrono::high_resolution_clock::now();           \
+    string __func_name = (FUNCTION_NAME);                               \
+    string __remarks = (REMARKS);\
+    printf("<TID: %u> Starting Time (%lld) for %s %s \n", static_cast<uint32_t>(gettid()), \
+    (long long)__start_time.time_since_epoch().count(), __func_name.c_str(), __remarks.c_str());
+
+#define STOP_TIME_AND_PRINT_DURATION                                         \
+    do {                                                                      \
+        auto __end_time = std::chrono::high_resolution_clock::now();          \
+        auto __duration = std::chrono::duration_cast<std::chrono::milliseconds>( \
+            __end_time - __start_time                                        \
+        ).count();                                                            \
+        printf("<TID:%u >Time for %s %s = %lld ms\n",                                  \
+               static_cast<uint32_t>(gettid()), __func_name.c_str(), __remarks.c_str(),                        \
+               (long long)__duration);                                        \
+    } while(0);
+
+
 #endif // __PORTABILITY_H
diff --git a/Source/plugins/MetaData.h b/Source/plugins/MetaData.h
index 206b4f6d..352b9abb 100644
--- a/Source/plugins/MetaData.h
+++ b/Source/plugins/MetaData.h
@@ -442,14 +442,20 @@ namespace Plugin {
             const std::initializer_list<subsystem>& precondition,
             const std::initializer_list<subsystem>& termination,
             const std::initializer_list<subsystem>& control)
-            : _version((major << 16) | (minor << 8) | patch)
+            : _startTime(std::chrono::high_resolution_clock::now())
+            , _version((major << 16) | (minor << 8) | patch)
             , _Id(Core::ClassNameOnly(typeid(ACTUALSERVICE).name()).Text())
             , _precondition(precondition)
             , _termination(termination)
             , _control(control) {
+            auto endTime = std::chrono::high_resolution_clock::now();
+            auto initDuration = std::chrono::duration_cast<std::chrono::milliseconds>(endTime - _startTime).count();
+            printf("Time for Metadata Initializing %s = %lld ms startTimefromEpoch(%lld)\n", _Id.c_str(), (long long)initDuration, (long long)_startTime.time_since_epoch().count());
+            START_TIME("Metadata Ctor", _Id.c_str());
             ASSERT(Core::System::ROOT_META_DATA == nullptr);
             Core::System::ROOT_META_DATA = this;
             Core::ServiceAdministrator::Instance().Register(this, &_factory);
+            STOP_TIME_AND_PRINT_DURATION
         }
         ~Metadata() {
             Core::ServiceAdministrator::Instance().Unregister(this, &_factory);
@@ -483,6 +489,7 @@ namespace Plugin {
         }
 
     private:
+        std::chrono::time_point<std::chrono::high_resolution_clock> _startTime;
         uint32_t _version;
         string _Id;
         std::vector<subsystem> _precondition;
