From: Sergey Borushevsky <Sergey_Borushevsky2@cable.comcast.com>
Date: Wed, 15 May 2022 02:50:32 +0200
Subject: [PATCH] WorkerPool_Increase.patch
RDK-45390: Added patch to increase queueSize for WorkerPool
To avoid full queue in WorkerPool when systemd makes over 30 parallel requests to activate plugins

Upstream-Status: Pending
Signed-off-by:  Sergey Borushevsky <Sergey_Borushevsky2@cable.comcast.com>
---
diff -upr a/Source/WPEFramework/PluginServer.h b/Source/WPEFramework/PluginServer.h
--- a/Source/WPEFramework/PluginServer.h	2024-05-08 02:48:58.157150418 +0300
+++ b/Source/WPEFramework/PluginServer.h	2024-05-08 17:12:03.896421445 +0300
@@ -125,7 +125,7 @@ namespace PluginHost {
             WorkerPoolImplementation& operator=(const WorkerPoolImplementation&) = delete;
 
             WorkerPoolImplementation(const uint32_t stackSize)
-                : Core::WorkerPool(THREADPOOL_COUNT, stackSize, 16, &_dispatch, this)
+                : Core::WorkerPool(THREADPOOL_COUNT, stackSize, 50, &_dispatch, this)
                 , _dispatch()
             {
                 Run();
