diff --git a/reference/src/barton-core-reference-app.c b/reference/src/barton-core-reference-app.c
index af88e88..ba5be10 100644
--- a/reference/src/barton-core-reference-app.c
+++ b/reference/src/barton-core-reference-app.c
@@ -179,7 +179,7 @@ static void completionCallback(const char *buf, linenoiseCompletions *lc)
 static gchar *getDefaultConfigDirectory(void)
 {
     struct passwd *pw = getpwuid(getuid());
-    gchar *confDir = stringBuilder("%s/.brtn-ds", pw->pw_dir);
+    gchar *confDir = stringBuilder("/opt/.brtn-ds");
     return confDir;
 }
 
@@ -220,6 +220,9 @@ static void setDefaultParameters(BCoreInitializeParamsContainer *params)
         b_core_property_provider_set_property_string(
             propProvider, B_CORE_BARTON_MATTER_SERIAL_NUMBER, "SN-123456789");
 
+        b_core_property_provider_set_property_string(
+            propProvider, "device.matter.environment", "development");
+
         // Set Manufacturing Date to "now"
         time_t now = time(NULL);
         struct tm *tm = localtime(&now);
