From 34c42740270e0d6cb519a28724a3e6e8b89101ad Mon Sep 17 00:00:00 2001
From: Sergey Borushevsky <Sergey_Borushevsky2@cable.comcast.com>
Date: Sat, 19 Jun 2021 17:45:04 +0000
Subject: RDKTV-31430 - Added TV related deps for DisplaySettings.
Source: COMCAST
Upstream-Status: N/A
Signed-off-by: Sergey Borushevsky <Sergey_Borushevsky2@cable.comcast.com>

diff -uprN a/wpeframework-displaysettings.service b/wpeframework-displaysettings.service
--- a/wpeframework-displaysettings.service	2024-07-09 23:22:54.827863978 +0300
+++ b/wpeframework-displaysettings.service	2024-07-10 19:27:36.708407535 +0300
@@ -1,7 +1,7 @@
 [Unit]
 Description=WPEFramework DisplaySettings Initialiser
-Requires=dsmgr.service pwrmgr.service wpeframework.service
-After=dsmgr.service pwrmgr.service wpeframework.service
+Requires=dsmgr.service pqserver.service wpeframework-hdmicecsink.service pwrmgr.service wpeframework.service
+After=dsmgr.service pqserver.service wpeframework-hdmicecsink.service pwrmgr.service wpeframework.service
 [Service]
 Type=oneshot
 RemainAfterExit=yes
