From 096d2a5038260827c841b038698a649b17ffaa21 Mon Sep 17 00:00:00 2001
From: Sergey Borushevsky <Sergey_Borushevsky2@cable.comcast.com>
Date: Sat, 19 Jun 2021 17:45:04 +0000
Subject: [PATCH] RDKTV-31430 - Added TV related deps for DisplaySettings.

---
 wpeframework-displaysettings.service | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/wpeframework-displaysettings.service b/wpeframework-displaysettings.service
index f11ad8c..decaff7 100644
--- a/wpeframework-displaysettings.service
+++ b/wpeframework-displaysettings.service
@@ -1,7 +1,7 @@
 [Unit]
 Description=WPEFramework DisplaySettings Initialiser
-Requires=wpeframework.service dsmgr.service wpeframework-powermanager.service
-After=wpeframework.service dsmgr.service wpeframework-powermanager.service
+Requires=wpeframework.service dsmgr.service pqserver.service wpeframework-hdmicecsink.service wpeframework-powermanager.service
+After=wpeframework.service dsmgr.service pqserver.service wpeframework-hdmicecsink.service wpeframework-powermanager.service
 PartOf=wpeframework.service
 [Service]
 Type=oneshot
-- 
2.43.0

