diff --git a/Source/com/IUnknown.cpp b/Source/com/IUnknown.cpp
index 7694a30c..e8e9537e 100644
--- a/Source/com/IUnknown.cpp
+++ b/Source/com/IUnknown.cpp
@@ -110,6 +110,15 @@ namespace ProxyStub {
 
                 if (result == Core::ERROR_TIMEDOUT) {
                     SYSLOG(Logging::Error, (_T("IPC method Invoke failed due to timeout (Interface ID 0x%X, Method ID 0x%X). Execution of code may or may not have happened. Side effects are to be expected after this message"), message->Parameters().InterfaceId(), message->Parameters().MethodId()));
+                    Dl_info info;
+                    void* vtable_addr = *reinterpret_cast<void**>(this);
+                    dladdr(vtable_addr, &info);
+                    fprintf(stderr, "pid=%d tid=%d %s fname=%s\n",
+                            getpid(),
+                            gettid(),
+                            __FUNCTION__,
+                            info.dli_fname);
+                    std::abort();
                 }
 
                 result |= COM_ERROR;
diff --git a/Source/websocket/JSONRPCLink.h b/Source/websocket/JSONRPCLink.h
index ed8026b2..65385d30 100644
--- a/Source/websocket/JSONRPCLink.h
+++ b/Source/websocket/JSONRPCLink.h
@@ -522,6 +522,15 @@ namespace WPEFramework {
 				, _scheduledTime(0)
 				, _versionstring()
 			{
+                Dl_info info;
+                void* vtable_addr = *reinterpret_cast<void**>(this);
+                dladdr(vtable_addr, &info);
+                fprintf(stderr, "pid=%d tid=%d %s callsign=%s fname=%s\n",
+                        getpid(),
+                        gettid(),
+                        __FUNCTION__,
+                        _callsign.c_str(),
+                        info.dli_fname);
 				if (localCallsign == nullptr) {
 					static uint32_t sequence;
 					_localSpace = string("temporary") + Core::NumberType<uint32_t>(Core::InterlockedIncrement(sequence)).Text();
@@ -556,6 +565,15 @@ namespace WPEFramework {
 			}
 			virtual ~LinkType()
 			{
+                Dl_info info;
+                void* vtable_addr = *reinterpret_cast<void**>(this);
+                dladdr(vtable_addr, &info);
+                fprintf(stderr, "pid=%d tid=%d %s callsign=%s fname=%s\n",
+                        getpid(),
+                        gettid(),
+                        __FUNCTION__,
+                        _callsign.c_str(),
+                        info.dli_fname);
 				_channel->Unregister(*this);
 
 				for (auto& element : _pendingQueue) {
