diff -upr a/apis/UserSettings/IUserSettings.h b/apis/UserSettings/IUserSettings.h
--- a/apis/UserSettings/IUserSettings.h	2025-03-24 17:34:06.559410109 +0200
+++ b/apis/UserSettings/IUserSettings.h	2025-03-29 02:58:15.604527643 +0200
@@ -74,6 +74,11 @@ struct EXTERNAL IUserSettings : virtual
         // @param service: the changed preferredClosedCaptionService.
         virtual void OnPreferredClosedCaptionServiceChanged(const string& service) {};
 
+        // @text onPrivacyModeChanged
+        // @brief The PrivacyMode setting has changed.
+        // @param privacyMode: "SHARE", "DO_NOT_SHARE".
+        virtual void OnPrivacyModeChanged(const string& privacyMode /* @text privacyMode */) {};
+
         // @alt onPinControlChanged
         // @brief The PinControl setting has changed.
         // @param pinControl: pinControl enabled or not.
@@ -213,6 +218,18 @@ struct EXTERNAL IUserSettings : virtual
     // @param service: Identifies the service to display e.g. "CC3".
     virtual uint32_t GetPreferredClosedCaptionService(string &service /* @out */) const = 0;
 
+    // @text setPrivacyMode
+    // @brief Sets the PrivacyMode.
+    // @details The setting should be honored by the Telemetry.
+    // If privacyMode is "DO_NOT_SHARE", logs and crash report should not be uploaded.
+    // @param privacyMode: "SHARE", "DO_NOT_SHARE"
+    virtual uint32_t SetPrivacyMode(const string& privacyMode /* @in @text privacyMode*/) = 0;
+
+    // @text getPrivacyMode
+    // @brief Gets the current PrivacyMode setting.
+    // @param privacyMode: "SHARE"
+    virtual uint32_t GetPrivacyMode(string &privacyMode /* @out @text privacyMode */) const = 0;
+
     // @alt setPinControl
     // @brief Sets PinControl ON/OFF. Parental Control as a whole is enabled or disabled.
     // @param pinControl: Enabled/Disabled
