Date: Apr 16 12:00:00 2020
From: 81a8aa1ff416750f3acb82a482d970a14ae2274f 
Subject: Added dbus env variable
Source: COMCAST
Upstream-Status: Pending
Signed-off-by: acheri988 <Anooj_Cheriyan@comcast.com>

diff --git a/conf/dsmgr.service b/conf/dsmgr.service
index d89a589..8b5344e 100644
--- a/conf/dsmgr.service
+++ b/conf/dsmgr.service
@@ -23,6 +23,7 @@
 
 [Service]
 Type=notify
+Environment=DBUS_SESSION_BUS_ADDRESS=unix:path=/var/run/dbus/system_bus_socket
 ExecStartPre=/bin/mkdir -p /opt/persistent/ds
 ExecStart=/usr/bin/dsMgrMain
 
