Date: Feb 07 20:41:00 2020
From: 8de3eeb24d7f2bcff2163cf1c112922046f4b6ea Jan 28 22:53:46 2021
Subject: Add essos resource manager env dependencies
Upstream-Status: Pending
Signed-off-by: Deekshit Devadas <deekshit.devadas@sky.uk>
---

Index: git/conf/dsmgr.service
===================================================================
--- git.orig/conf/dsmgr.service
+++ git/conf/dsmgr.service
@@ -24,6 +24,8 @@
 [Service]
 Type=notify
 Environment=DBUS_SESSION_BUS_ADDRESS=unix:path=/var/run/dbus/system_bus_socket
+Environment="XDG_RUNTIME_DIR=/tmp"
+Environment="HDMIIN_USE_ESSRMGR=1"
 ExecStartPre=/bin/mkdir -p /opt/persistent/ds
 ExecStart=/usr/bin/dsMgrMain
 
