Date: Apr 26 12:59:00 2023
From: 14cb753b51b668d21d216be299d60eb7414ef7c0 Apr 7 07:47:30 2023 +0000
Subject: Add dsmgr service env settings
Upstream-Status: Pending
Signed-off-by: Deekshit Devadas <deekshit.devadas@sky.uk>

Index: git/conf/dsmgr.service
===================================================================
--- git.orig/conf/dsmgr.service
+++ git/conf/dsmgr.service
@@ -26,6 +26,8 @@ Type=notify
 Environment=DBUS_SESSION_BUS_ADDRESS=unix:path=/var/run/dbus/system_bus_socket
 Environment="XDG_RUNTIME_DIR=/tmp"
 Environment="HDMIIN_USE_ESSRMGR=1"
+ExecStartPre=/bin/sh -c 'DS_AUDIO_PARAM_MAP_INI=`/usr/bin/panelIDConfig -v`;/bin/systemctl set-environment DS_AUDIO_PARAM_MAP=$DS_AUDIO_PARAM_MAP_INI'
+ExecStartPre=/bin/sh -c 'AUDIO_PROFILE_INI=`/usr/bin/panelIDConfig -r`;/bin/systemctl set-environment MS12_AUDIO_PROFILE=$AUDIO_PROFILE_INI'
 ExecStartPre=/bin/mkdir -p /opt/persistent/ds
 ExecStart=/usr/bin/dsMgrMain
 
