From: Stefan Verkoyen <sverkoyen.contractor@libertyglobal.com>
Subject: RDKCMF-8791 reference image: initial support for LISA
Index: git/LISA/LISA.config
===================================================================
--- git.orig/LISA/LISA.config
+++ git/LISA/LISA.config
@@ -1,16 +1,13 @@
 set(autostart true)

 map()
-    kv(dbpath "/mnt/apps/dac/db")
-    kv(appspath "/mnt/apps/dac/images")
-    kv(datapath "/mnt/data/dac")
-    kv(annotationsFile "annotations.json")
+    kv(dbpath "/opt/dac_apps")
+    kv(appspath "/opt/dac_apps/apps")
+    kv(datapath "/opt/dac_apps/data")
+    kv(annotationsFile "config.json")
     kv(annotationsRegex "public\\.*")
-    key(root)
-    map()
-        kv(outofprocess true)
-        kv(mode "Container")
-    end()
+    kv(downloadRetryMaxTimes 6)
+    kv(downloadRetryAfterSeconds 30)
 end()

 ans(configuration)
Index: git/LISA/LISA.conf.in
===================================================================
--- git.orig/LISA/LISA.conf.in
+++ git/LISA/LISA.conf.in
@@ -1,12 +1,10 @@
 autostart = "true"

 configuration = JSON()
-configuration.add("dbpath", "/mnt/apps/dac/db")
-configuration.add("appspath", "/mnt/apps/dac/images")
-configuration.add("datapath", "/mnt/data/dac")
-configuration.add("annotationsFile", "annotations.json")
+configuration.add("dbpath", "/opt/dac_apps")
+configuration.add("appspath", "/opt/dac_apps/apps")
+configuration.add("datapath", "/opt/dac_apps/data")
+configuration.add("annotationsFile", "config.json")
 configuration.add("annotationsRegex", "public\\.*")
-rootobject = JSON()
-rootobject.add("outofprocess", "true")
-rootobject.add("mode", "Container")
-configuration.add("root", rootobject)
+configuration.add("downloadRetryMaxTimes", "6")
+configuration.add("downloadRetryAfterSeconds", "30")
