diff --git a/apis/OpenCDMi/IDRM.h b/apis/OpenCDMi/IDRM.h
index 2cc86dd..329b48a 100644
--- a/apis/OpenCDMi/IDRM.h
+++ b/apis/OpenCDMi/IDRM.h
@@ -269,6 +269,12 @@ typedef struct {
     SubSampleInfo*     subSample;       // SubSample mapping - Repeating pair of Clear bytes and Encrypted Bytes representing each subsample.
 } SampleInfo;
 
+//FPS
+typedef struct {
+  uint32_t offset;
+  uint32_t size;
+} Cdmi_FPSSliceInfo;
+
 // IStreamProperties to provide information about the current stream
 class IStreamProperties {
 public:
@@ -384,6 +390,12 @@ PUSH_WARNING(DISABLE_WARNING_DEPRECATED_USE)
 POP_WARNING()
     }
 
+    virtual CDMi_RESULT Decrypt(uint64_t MovieId,uint64_t CryptorId, uint32_t ContentTyp, uint8_t *encryptedData, uint32_t encryptedDataSize, const Cdmi_FPSSliceInfo *SliceInfoArray,uint32_t SliceInfoArrayCount, uint8_t *iv, void *svphan)
+    {
+	    return (CDMi_METHOD_NOT_IMPLEMENTED);
+    }
+
+
     virtual CDMi_RESULT ReleaseClearContent(
         const uint8_t* f_pbSessionKey,
         uint32_t f_cbSessionKey,
@@ -447,6 +459,47 @@ public:
     virtual CDMi_RESULT DestroyMediaKeySession(
         IMediaKeySession* f_piMediaKeySession)
         = 0;
+
+    // fairplay
+    virtual CDMi_RESULT CreateMovieSession(uint32_t version, const uint8_t* cert, uint32_t certificatesize,
+        const uint32_t* versionlist, uint32_t versionListSize, uint64_t* movieIdOut)
+    {
+        return CDMi_METHOD_NOT_IMPLEMENTED;
+    }
+    virtual CDMi_RESULT DestroyMovieSession(uint32_t version, uint64_t movieId)
+    {
+        return CDMi_METHOD_NOT_IMPLEMENTED;
+    }
+    virtual CDMi_RESULT GenerateChallengeWithVersionList(uint32_t version, uint64_t movieId,
+        const uint8_t* assetidData, uint32_t assetidSize, const uint32_t* versionlist, uint32_t versionListSize,
+        const uint8_t* streamerChallengeData, uint32_t streamerChallengeSize,
+        uint64_t cryptorId, uint8_t* licenseChallengeBuffer, uint8_t* licenseSize, uint8_t* session)
+    {
+        return CDMi_METHOD_NOT_IMPLEMENTED;
+    }
+    virtual CDMi_RESULT DestroyServerExchange(uint32_t version, uint8_t* session)
+    {
+        return CDMi_METHOD_NOT_IMPLEMENTED;
+    }
+    virtual CDMi_RESULT ProcessLicense(uint32_t version, uint8_t* session,
+        const uint8_t* licenseData, uint32_t licenseDataSize, uint8_t* cryptoId)
+    {
+        return CDMi_METHOD_NOT_IMPLEMENTED;
+    }
+    virtual CDMi_RESULT InitLibrary()
+    {
+        return CDMi_METHOD_NOT_IMPLEMENTED;
+    }
+
+     virtual CDMi_RESULT DestroyCryptor(uint32_t version, uint64_t cryptoId)
+     {
+             return CDMi_METHOD_NOT_IMPLEMENTED;
+     }
+
+    virtual CDMi_RESULT GetLibraryVersion(uint32_t* versionOut)
+    {
+        return CDMi_METHOD_NOT_IMPLEMENTED;
+    }
 };
 
 // IMediaKeySession defines the MediaKeySessionExt interface.
