diff --git a/apis/LifecycleManager/ILifecycleManager.h b/apis/LifecycleManager/ILifecycleManager.h
index c9b9488..5e25bba 100755
--- a/apis/LifecycleManager/ILifecycleManager.h
+++ b/apis/LifecycleManager/ILifecycleManager.h
@@ -56,7 +56,7 @@ struct RuntimeConfig {
 #define RUNTIME_CONFIG
 #endif
 
-// @text:keep
+// @json 1.0.0 @text:keep
 struct EXTERNAL ILifecycleManager : virtual public Core::IUnknown {
 
     enum LifecycleState : uint8_t {
@@ -108,9 +108,11 @@ struct EXTERNAL ILifecycleManager : virtual public Core::IUnknown {
     virtual Core::hresult SpawnApp(const string& appId /* @in */, const string& launchIntent /* @in */, const LifecycleState targetLifecycleState /* @in */, const RuntimeConfig& runtimeConfigObject /* @in */, const string& launchArgs /* @in */, string& appInstanceId /* @out */, string& errorReason /* @out */, bool& success /* @out */) = 0;
 
     /** Get the list of loaded applications */
-    // @json:omit
     // @text setTargetAppState
     // @brief Set the state for the application instance
+    // @param appInstanceId: Instance ID of the application
+    // @param targetLifecycleState: Target lifecycle state to set
+    // @param launchIntent: Launch intent to use
     virtual Core::hresult SetTargetAppState(const string& appInstanceId /* @in */, const LifecycleState targetLifecycleState /* @in */, const string& launchIntent /* @in */) = 0;
 
     /** unload the application */
