Subject:LLAMA-8846: Force center textAlign for all Sky TTML
Upstream-Status: N/A
Source: Sky
Signed-off-by: Stephen Waddell <stephen.wadell@sky.uk>

diff --git a/src/Parser/StyleSet.cpp b/subttxrend-ttml/src/Parser/StyleSet.cpp
index f6ada7e..983f726 100644
--- a/src/Parser/StyleSet.cpp
+++ b/src/Parser/StyleSet.cpp
@@ -35,6 +35,7 @@
 #include <string>
 #include <utility>
 #include <sstream>
+#include <subttxrend/common/Logger.hpp>
 
 namespace subttxrend
 {
@@ -195,6 +196,9 @@
     return displayAlign;
 }
 
+common::Logger logger("TtmlEngine", "StyleSet");
+
+
 } // namespace anonymous
 
 const ColorArgb& StyleSet::getColor() const
@@ -289,7 +293,8 @@ void StyleSet::parseAttribute(const std::string& name,
             m_fontSize = sizeResult.size;
         }
     } else if (name == "textAlign") {
-        m_textAlign = parseTextAlign(value);
+        logger.info("%s LLAMA-8155: Ignoring incoming textAlign property (%s) - defaulting to \"center\"", __LOGGER_FUNC__, value.c_str());
+        //m_textAlign = parseTextAlign(value);
     } else if (name == "displayAlign") {
         m_displayAlign = parseDisplayAlign(value);
     } else if (name == "fontFamily") {
