From 62f8c1abb27ccd53e06eff12d5ae630a21cd709d Mon Sep 17 00:00:00 2001
From: Ansu Mathew <ansu.mathew@rdkcentral.com>
Date: Mon, 6 Jul 2026 12:51:58 +0000
Subject: [PATCH] Remove rdk-arm from starboard build script

---
 starboard/build/platforms.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/starboard/build/platforms.py b/starboard/build/platforms.py
index a8159ae1b5f..efecd923a55 100644
--- a/starboard/build/platforms.py
+++ b/starboard/build/platforms.py
@@ -38,7 +38,6 @@ PLATFORMS = {
     'android-x86': 'starboard/android/x86',
     'raspi-2': 'starboard/raspi/2',
     'raspi-2-skia': 'starboard/raspi/2/skia',
-    'rdk-arm': 'starboard/contrib/rdk/src/third_party/starboard/rdk/arm',
     'evergreen-x64': 'starboard/evergreen/x64',
     'evergreen-arm-hardfp': 'starboard/evergreen/arm/hardfp',
     'evergreen-arm-softfp': 'starboard/evergreen/arm/softfp',
-- 
2.44.4

