diff --git a/starboard/nplb/BUILD.gn b/starboard/nplb/BUILD.gn
index ca1f90be780..45482a7928e 100644
--- a/starboard/nplb/BUILD.gn
+++ b/starboard/nplb/BUILD.gn
@@ -16,6 +16,7 @@ import("//starboard/build/config/os_definitions.gni")
 
 declare_args() {
   sb_enable_cast_codec_tests = false
+  enable_test_data_download = false
 }
 
 config("cast_codec_tests") {
@@ -342,9 +343,13 @@ target(gtest_target_type, "nplb") {
   data_deps = [
     "//cobalt/network:copy_ssl_certificates",
     "//starboard/nplb/testdata/file_tests:nplb_file_tests_data",
-    "//starboard/shared/starboard/player:player_download_test_data",
   ]
 
+
+  if (enable_test_data_download) {
+    data_deps += [ "//starboard/shared/starboard/player:player_download_test_data" ]
+  }
+
   if (is_clang_16) {
     # signedness_and_size_of_enum_test.cc casts -1 to enum value
     cflags = [ "-Wno-enum-constexpr-conversion" ]
